# Introduction {#intro}
\pagenumbering{arabic}
\setcounter{page}{1}

## Background

<p> There are several techniques used for studying genetics and mapping the results.  Some of the more populor techniques include cross-breeding experiments or, in the case of  humans, the examination of family histories, known as pedigrees. More recently, CRISPR/Cas9 can be used to mimic mitotic recombination to help map out genes as well. (\cite{sadhu2016crispr}) </p>

<p> Construction of genetic maps are a variety of techniques used to show relative positions between genes or other sequence features of the genome and the phenotype that is controlled by such sequences. Genes are very useful markers but they are by no means ideal. One problem, especially with larger genomes such as those of vertebrates and flowering plants, is that a map based entirely on genes is not very detailed.(\cite{brown2006genomes}) Genes have long areas of non-coding regions between them and therefore result in large gaps from gene to gene.  This is further complicated because not every gene has allelic forms that can be easily or conventiently distinguished.  With these considerations in mind gene maps may not be comprehensive enough and other markers may be needed. </p>

<p> According to brown, mapped features that are not genes are called DNA markers. As with gene markers, a DNA marker must have at least two alleles to be useful. There are three types of DNA sequence feature that satisfy this requirement: restriction fragment length polymorphisms (RFLPs), simple sequence length polymorphisms (SSLPs), and single nucleotide polymorphisms (SNPs). (\cite{brown2006genomes})  The genetic markers that have been emphaised in this work are single nucleotide polymorphisms.  Attempting to be at the highest levels of resolution for identifying quantitative traits, using SNPs are the most specific case.  This will give exact location of the nucletid that may be impacting the genetic control over the phenotype. </p>


```{r Snp-Pic, fig.cap='SNP Picture', out.width='80%', fig.asp=.75, fig.align='center', echo = FALSE}
knitr::include_graphics("images/SNP_Picture.png")
```

<p> There are several goals to genetic mapping and association studies that identify certain regions of the genome that contain genes involved in specifying a quantitative trait, refered to as quantitative trait loci (QTLs).  One main goal is to estimate the genetic effects of these loci. The relationship between the genetic effects of QTLs and the phenotypic value of quantitative traits can be described by a linear model (\cite{collard2005introduction}, \cite{xu2007empirical}). Typically, because of the high throughput nature of the data there are a large number of markers across the whole genome, and most of the markers may have very little or next no effect on the phenotype under study. The models can be very sparse, with most cases, the number of genetic markers or variables is bigger than the sample size, especially when interactions among markers are considered. This makes a model is oversaturated and furher model selection techniques may be required to capture the necessary information. \cite{dong2015accurate} </p>



## Some Exisiting Methods

Numerous methods exist and are being developed to measure and find quantitative trait loci (QTL) effects.  These methods can broadly fall into three main categories.  These categories are Least-Square methods, maximum likelihood and Bayesian approaches.  (\cite{wu2007statistical})  Each method has advantages and considerations that you would need to be aware before conducting analyses to find QTL effects from the given markers.  A brief discussions on a few of the methods are given to highlight some areas of consideration and how the methods proposed can handle such considerations.  

Marker Regression would fall in the category of Least Squares approaches.  If looking at one marker analysis general t-test and ANOVA procedures can be used to analyze the relationship.  It is not recommended however for use in general practice because you do not know how dense the markers are measured.  QTL interval mapping would be preferred in such an analysis because the methods take account for missing genotype data that may not have been measured.  When estimating a QTL position through maximum likelihood methods, like interval mapping, positions of other possible QTLs could affect the detection of the true position.  Neighboring QTLs could possibly flatten the likelihood in instances where there are multiple QTLs on the same chromosome. This would make an effect look less significant at a given location than it actually is.  Another possibility is that in the search over the interval you may find an area where the likelihood could reach a peak but could be a “ghost” QTL.  This is where an effect is observed because a neighboring QTL is skewing the results at the particular position you are looking in and the result is a false discovery of the position.  Marker Regression has been shown to improve interval mapping, which is call Composite Interval Mapping.  This is where the QTL position found is also combined in a linear regression where the covariates are the other markers in the dataset.  By including the markers as covariates the other position in the chromosome are accounted for in the analysis and false discovery is reduced.  

<p> The analysis of interval mapping and single marker analyses has shown to be effective but it limits our inference to one marker at a time as a possible loci that controls a trait.   Using Marker Regression however you can incorporate multiple markers in a single analysis to test for possible QTL for a given trait.  It is cautioned that running such an analysis is only an approximate test because the null hypothesis is there is no difference between the marker levels and therefore a non-mixture distribution but the alternative is a mixture of distributions.  The assumptions regression would make of the errors within the marker type to be normally distributed may not be entirely met if the QTL’s fall between the marker regions.  However \cite{whittaker1996mapping} have shown that a direct regression of phenotypes on marker types, provides the same information about location of QTL-effects without having to step to all positions on the interval.  With this information using the entire marker set in a regression analysis would provide a nice, computationally efficient way to map out the genetic architecture of a trait.</p>


##	Chapter Overview

The main goal of this paper is to propose an improved variational linear regression approach for high scale variable selection problems such as the ones arising in epistatic analysis

The variable selection procedure for QTLs mapping can be seen as one of deciding which subset of variables have effects on phenotypes, and identifying out all possible effects of those markers.


### iForm (change chapter names)

### iForm Higher Order (change chapter names)
 
### iForm Funcional Mapping (change chapter names)










```{r system-map, fig.cap='Systems Map', out.width='80%', fig.asp=.75, fig.align='center', echo = FALSE}
knitr::include_graphics("images/SystemsMapping.png")
```





You can label chapter and section titles using `{#label}` after them, e.g., we can reference Chapter \@ref(intro). If you do not manually label them, there will be automatic labels anyway, e.g., Chapter \@ref(methods).

Figures and tables with captions will be placed in `figure` and `table` environments, respectively.

```{r nice-fig, fig.cap='Here is a nice figure!', out.width='80%', fig.asp=.75, fig.align='center'}
par(mar = c(4, 4, .1, .1))
plot(pressure, type = 'b', pch = 19)
```

Reference a figure by its code chunk label with the `fig:` prefix, e.g., see Figure \@ref(fig:nice-fig). Similarly, you can reference tables generated from `knitr::kable()`, e.g., see Table \@ref(tab:nice-tab).

```{r nice-tab, tidy=FALSE}
knitr::kable(
  head(iris, 20), caption = 'Here is a nice table!',
  booktabs = TRUE
)
```

You can write citations, too. For example, we are using the **bookdown** package [@R-bookdown] in this sample book, which was built on top of R Markdown and **knitr** [@xie2015].
