# iForm Functional Mapping (A computational method)


## Motivation

 Growth and developmental traits can mostly be better described by a functional process (\cite{hernandez2015understanding}, \cite{muraya2017genetic}), it is more biologically meaningful to map these traits as growth curves (\cite{sun2015mapping}). There have been a few different approaches that have integrated growth equations into genetic mapping via the likelihood function, leading to the birth of a so-called functional mapping models (\cite{ma2002functional}; \cite{wu2006functional}; \cite{li2015dynamic}; \cite{muraya2017genetic}). These style appoaches can allow for the developmental change of genetic control to be characterized across both time and as well as space (\cite{he2010mapping}; \cite{li2010functional}). 


 **Shoot-Root Paper**
  (The genetic architecture of shoot-root covariation during seedling emergence of a dessert tree, Populus euphratica)
 
 By modeling the longitudinal mean-covariance structures using a set of parsimonious parameters, functional mapping has proven of great statistical power in gene identification and the utilization of sparse phenotypic data (Hou et al., 2005, 2006). An alternative to functional mapping is to map growth QTLs by estimating growth parameters for each genotype based on growth equations and associating these parameters with markers (Wu et al., 2002). 
 
 
 
 From fundamental principles of biophysical and biochemical processes, logistic equations that capture different stages of organ development have been derived, which show robust biological relevance (West et al., 2001). Sun et al. (2014) dissect logistic growth curves into several key landmarks of development using the concept of heterochrony, defined as the asymptotic growth, relative growth rate, the timing of inflection and the duration of linear growth. 
 

  **from research statement**

Treating the molecular biology of an organism as a complex trait, it would be likely that the expression profiles of certain genes, proteins or other metabolites would follow a more functional or dynamic phenotype. This information could be lost or greatly limited by treating the response as a single static predictor. In an attempt to capture all relevant information available I am currently extending the procedure further to include the entire growth function as a phenotype. The mean growth curve is estimated for the sample and then orthogonal polynomials are used to assist in fitting the genetic effects for each marker or interaction between the markers. This would allow the genetic effect some flexibility over time and give a more representative fit. I anticipate submitting a third manuscript in late spring around this topic. This could also be used with other semi parametric functions to model dynamics or other types of non-linear functions that characterizes the biological systems being evaluated.


**Update**
  (This is from Han Hao's proposal, update for yours) 

Changes in developmental timing and rate, named as heterochrony, have long
been believed to be a major force in the evolution of phenotype (\cite{gould1977ontogeny}; \cite{wilson1988heterochrony}). It has been observed that relatively few genetic changes
in heterochrony through the endocrine regulation of metamorphosis can cause
profound morphological consequences (Moss, 2007). For example, although humans
and chimpanzees are closely related, their skull shape and brain growth are different
dramatically during early development (Rice, 2002; Mitteroecker et al., 2004; King,
2004). A recent phylogenic investigation revealed that changes in developmental
timing are a crucial step for birds to evolve from dinosaurs (Bhullar, 2012). The
consequence of these changes leads birds to take months to reach sexual maturity,
allowing them to retain the physical characteristics of baby stages characterized by dinosaurs that take years to mature. One question that naturally arises from these
evolutionary divergences is what mechanisms are implicated for heterochrony and
the change of biological clock.


Many studies have pursued to identify the molecular control of developmental
timing; mostly using Caenorhabditis elegans as an example, these studies have
identified heterochronic genes that orchestrate the timing of cell divisions and fates
during development regulated by microRNAs and their targets (Ambros, 2000;
Rougvie, 2001; Pasquinelli and Ruvkun, 2002; Banerjee and Slack, 2002; Moss, 2007).
Focusing on particular pathways causing heterochonic changes, none of these studies
has provided an entire picture of the genetic control of heterochrnoy. Furthermore,
the effects of heterochronic genes on the evolution of complex phenotypes have not
been quantified, limiting the inference and prediction of evolutionary changes.


In this chapter, we develop a general framework for characterizing the genetic
architecture of heterochrony based on widely used genetic mapping approaches.
Genetic mapping has been proved to be powerful for mapping and studying quantitative
trait loci (QTLs) involved in complex traits (Lander and Botstein, 1986),
and has been increasingly integrated with network biology to better elucidate the
mechanisms of the way QTL acts and interacts with other factors (Wang et al.,
2012b). In the decade, genetic mapping has developed to a point at which this
approach can characterize QTLs that control the process of development, leading
to the birth of functional mapping (Ma et al., 2002; Wu and Lin, 2006; Li and
Wu, 2010). Functional mapping implements mathematical aspects of developmental
principles into a mapping framework, equipped with a capacity to study the
interplay between QTLs and development. Here, we extend functional mapping
to characterize QTLs controlling heterochrony (named hQTLs), specified by three
parameters (i) the onset of a particular process, (ii) its offset, and (iii) the rate at which the process proceeds. By using growth equation as an example, we exemplify the procedure of model derivations as well as the practical use of the model in hQTL detection. In the end, we discuss an issue of how hQTLs can be integrated
with developmental evolutionary biology (evo-devo), a fast-growing discipline of
biology in the recent years.



**Update**

GENERALIZED ADDITIVE MODEL SELECTION \cite{chouldechova2015generalized}

In many applications it may be too restrictive to suppose that the effect
of all of the predictors is captured by a simple linear fit of the form, $η(x_i) =  \sum_{j=1}^p \beta_jx_{ij}$.
Generalized additive models, introduced in \cite{hastie1990generalized}, allow for greater
flexibility by modeling the linear predictor of a generalized linear model as a sum of more
general functions of each variable: $$η(x_i) = \sum_{j=1}^p f_j(x_{ij}),$$
where the $f_j$ are unknown functions, assumed to be smooth or otherwise low-complexity.


Add a third paragraph to motivation section
  Continue on about the rest of the chapter as  an overview
  
  **2HIGWAS**
  
  In this study, we develop a high-dimensional varyingcoefficient
model to chart a complete picture of the genetic
architecture of complex traits that are dynamically expressed
on a time–space scale.



However, integrating all SNPs and functional mapping to systematically search for loci for developmental processes has not been explored thus far. This integration can not only construct a precise developmental genotype–phenotype map that cannot be produced separately by each approach, but also contains sophistication and challenges in technical development. First, such integration needs the resolution of how significant predictors can be chosen from a high-dimensional pool of SNPs under the sparsity assumption. Many statisticians have made tremendous effort to improve computational expediency, statistical accuracy and algorithmic stability in high dimensionality [22–28]. Second, given that phenotypic data are observed in a longitudinal time series, a step of embedding the multiple response structure in a biologically meaningful way is essential.






## Methods

### Functional Mapping

**Update**
  (This is from Han Hao's proposal, update for yours)
  
Functional mapping is a group of methods used for mapping QTLs related to functional valued traits that are measured over a certain time period. These function-valued traits are widely seen in growth analysis, shape analysis, network analysis, and clinical trials. By integrating functional features with genetic analysis, functional mapping methods often help to increase the statistical power and the biological relevance between the detected QTLs and biological traits. Functional mapping methods were first developed using parametric curves to describe the functional valued traits, such as growth curves or pharmacodynamic models (Ma et al., 2002; Lin et al., 2005). Later, semi-parametric and non-parametric models were introduced for complex traits that do not have specific mathematical forms (Das et al., 2011, 2013).


The key of functional mapping methods is the modelling of both functional means and the covariance structure across measurements from the same individual. When modelling the functional means, biological background is first examined to select the best function, either parametric or non-parametric. The covariance structure is usually modeled by parsimonious and flexible approaches such as autoregressive, antedependence, or nonparametric structures. The functional means and covariance structures can be integrated into a likelihood function, then hypothesis testing can be performed with a likelihood ratio testing approach.






**Update: Shoot-Root Paper**
 (pages 13 - 14)
fitted by a three-parameter growth equation, through a nonlinear least-square approach, which is expressed as
                                   (1)
where g(t) is the trait value at time t, and three parameters a, b and r have different biological meanings: a is the limit value of g when t→∞, r is the relative growth rate, and a/ (1 + b) denotes the initial value of g when t = 0. After the three growth parameters were estimated for each progeny, we further determined heterochronic parameters, i.e., the timing of inflection point (tI), the timing of maximum acceleration (ta), the timing of maximum deceleration (td), and the duration of linear growth (L) (Sun et al., 2014).



QTL mapping: For each progeny, we estimated a series of heterochronic parameters using equation (1) and then treated these estimates as phenotypic values to perform QTL mapping. There are two statistical approaches for QTL mapping, mixture model for sparse molecular markers and multiplicative model for dense markers. Because the linkage map constructed is quite dense, we employed the multiplicative model that assumes QTLs are located at the positions of markers. For the same heterochronic parameter expressed in shoot length and taproot length, the multiplicative likelihood model is expressed as
                         (2)
where  is the unknown parameters;$ yi = (y1i, y2i)$ is the growth parameter vector of progeny i for shoot length (coded by 1) and taproot length (coded by 2); nj is the number of progeny with SNP genotype j; and $fj(yi)$ is a bivariate normal distribution for progeny i with the expected mean vector for genotype j (1j, 2j) and the variance-covariance matrix  containing the variances (12, 22) and correlation between the two traits (). Statistical methods based on the likelihood (2) have been established to estimate the model parameters  = (1j, 2j, 12, 22, ).





### Regression by linear combination of  basis functions

By focusing on linear regression as a linear combination of basis functions

** GAM Selection paper**

Our proposed estimator selects between fitting each fj as zero, linear, or nonlinear, as
determined by the data. In so doing it retains the interpretability advantages of linear fits
when appropriate, while capturing strong non-linear relationships when they are present.
Our method, which we call GAMSEL (Generalized Additive Model Selection), is based on
optimizing a penalized (negative) log-likelihood criterion of the form


**Regression by linear combination of basis functions.  by Risi Kondor**

Solving for the θ’s
To find the optimal value for $θ_0, θ_1, . . . , θ_P$ we
1. define a loss function L;
2. using the loss function define the empirical risk $Remp(θ)$ quantifying the loss over all the training data for particular values of $θ0, θ1, . . . , θP$ ;
3. solve for the particular setting of the parameters (denoted $θ_0, θ_1, . . . , θ_P$)
that minimizes the empirical risk.

We shall use the squared error loss function
$$L(y, f(x)) =\frac{1}{2}(y − f(x))^2$$







### Legendre Polynomials

```{r Legendre-fig, fig.cap='First 10 Legendre Polynomials', out.width='80%', fig.asp=.75, fig.align='center', echo = FALSE}
library(ggplot2)
library(data.table)
library(RColorBrewer)

Legendre<-function( t, np.order=1,tmin=NULL, tmax=NULL )
{
  u <- -1
  v <- 1
  if (is.null(tmin)) tmin<-min(t)
  if (is.null(tmax)) tmax<-max(t)
  nt <- length(t)
  ti    <- u + ((v-u)*(t-tmin))/(tmax - tmin)
  np.order.mat <- matrix(rep(0,nt*np.order),nrow=nt)
  if(np.order >=1)
    np.order.mat[,1] <- rep(1,nt)
  if (np.order>=2)
    np.order.mat[,2] <- ti
  if (np.order>=3)
    np.order.mat[,3] <- 0.5*(3*ti*ti-1)
  if (np.order>=4)
    np.order.mat[,4] <- 0.5*(5*ti^3-3*ti)
  if (np.order>=5)
    np.order.mat[,5] <- 0.125*(35*ti^4-30*ti^2+3)
  if (np.order>=6)
    np.order.mat[,6] <- 0.125*(63*ti^5-70*ti^3+15*ti)
  if (np.order>=7)
    np.order.mat[,7] <- (1/16)*(231*ti^6-315*ti^4+105*ti^2-5)
  if (np.order>=8)
    np.order.mat[,8] <- (1/16)*(429*ti^7-693*ti^5+315*ti^3-35*ti)
  if (np.order>=9)
    np.order.mat[,9] <- (1/128)*(6435*ti^8-12012*ti^6+6930*ti^4-1260*ti^2+35)
  if (np.order>=10)
    np.order.mat[,10] <- (1/128)*(12155*ti^9-25740*ti^7+18018*ti^5-4620*ti^3+315*ti)
  if (np.order>=11)
    np.order.mat[,11] <- (1/256)*(46189*ti^10-109395*ti^8+90090*ti^6-30030*ti^4+3465*ti^2-63)
  return(np.order.mat)
}

L <- data.frame(t = seq(0, 1, by = 0.01), Legendre(seq(0, 1, by = 0.01), 10))
L_melt <- melt(L, id = "t")

ggplot(L_melt, aes(x = t, y = value, group = variable, color = brewer.pal(10, "Set3")[variable])) + 
  geom_line() + theme(legend.position="none")


```

**From Wikipedia**
An important property of the Legendre polynomials is that they are orthogonal with respect to the L2-norm on the interval −1 ≤ x ≤ 1:

$$P_n(x) = \frac{1}{2^n}\sum_{k=0}^{n}{{n}\choose{k}}^2(x-1)^{n-k}(x+1)^k$$

$$\sum_{k=0}^{n}{{n}\choose{k}}{{-n-1}\choose{k}}{\left(\frac{1-x}{2}\right)}^k$$
\begin{equation}
\begin{split}
P_n(x) & = \frac{1}{2^n}\sum_{k=0}^{n}{{n}\choose{k}}^2(x-1)^{n-k}(x+1)^k \\
& = \sum_{k=0}^{n}{{n}\choose{k}}{{-n-1}\choose{k}}{\left(\frac{1-x}{2}\right)}^k \\
& = 2^{-n}\sum_{k=0}^{n} x^k {{n}\choose{k}}{{\frac{n+k+1}{2}}\choose{k}} \\
\end{split}
(\#eq:leg-eq)
\end{equation}

**update** 2HIGWAS \cite{jiang20152higwas}

We consider the complex trait as a growth trait. Thus, it is biologically meaningful to implement a growth equation, like a logistic curve, to describe growth trajectory \cite{west2001general}. We describe the population mean growth curve by the growth equation 
$$ \mu(t) = a/(1 + b * exp(-r*t)) $$

where a, b and r are growth parameters each of biological interpretation, with a being the asymptotic growth, b being the initial amount of growth and r being the relative growth rate. Timevarying additive and dominant effects of significant SNPs are modeled by a nonparametric approach, such as Legendre orthogonal polynomial used in quantitative genetic studies (\cite{olori1999estimating},\cite{li2010functional}], expressed as

$$ \alpha_j(t) = (L_0(t), L_1(t), ... , L_s(t))*(u_{j0}, u_{j1},...,u_{js})^T $$
$$ \beta_j(t) = (L_0(t), L_1(t), ... , L_{s'}(t))*(v_{j0}, v_{j1},...,v_{js'})^T $$

where $L_0(t), L_1(t), ... , L_s(t)$ and  $L_0(t), L_1(t), ... , L_{s'}(t)$ are the LOP of orders $s$ and $s'$, respectively; and $u_{j0}, u_{j1},...,u_{js}$ and $v_{j0}, v_{j1},...,v_{js'}$ are the vectors of time-invariant additive and dominant effects, respectively. Orders $s$ and $s'$, selected from information criteria, such as Akaike information criterion (AIC) and Bayesian information criterion (BIC), are usually much smaller than M so that the dimension of response phenotypic data is reduced through LOP modeling. The variance maxtrix of residual errors is assumed to follow the first-order structured antedependence [SAD(1)] process (\cite{li2010functional}). The SAD(1) model has been used in previous growth modeling studies (\cite{li2010functional}, \cite{ahn2010functional}, \cite{das2011dynamic}).




**free write**

Because the non-parametic nature of the legendre orthogonal polynomials, it was advantages for both dimension reduction and also handling unevenly spaced, missing or non-uniform time measurements from different subjects in the dataset.  

The search performed by the procedure checks and uses the legendre polynomials of different orders to find the best fitting form of the genetic variation from the mean curve for each of the genotypes or epistasis between genotypes considered in the model.  



### Model 

**2HIGWAS**
Because interaction effects are considered,
the coverage probability P is redefined as three quantities:
PA, the coverage rate of all true predictors; PM, the coverage
rate of all main effects; and PI, the coverage rate of all interaction
effects.

__BIC__

Similarly, following the classical BIC rule (Swartz 1978), we can define the BIC rule for the cubic MESS model as

$$BIC(\lambda, \lambda_v ) = -2*LogLik + log(n)*(df + df_v)$$
n is the number of subjects \cite{wu2006nonparametric}
$$\lambda = smoothing paramter$$
$$\lambda_v = smoothing paramter, number of knots in the cubic spline$$


Considering a comprehensive gentotyping for n progeny each measure across different time points.  









## Application


### Simulation Studies

As statistical issues become more complex they are going to be more analytically intractable and computational methods will need to close that gap to show the effectiveness of new models and procedures.  


Extensive simulation studies were performed to ascertain the validity of the model.  Computational verified by cross validation, bootstrapping.  Having a training and a testing set to help provide some insight on false positive rates, selecting the correct fit of the polynomial for the genetic and epsitatic effects.  


* rates at which correct markers/epistasis was selected?
* rates at which the correct order polynomial was fit for the correct marker/epistasis



### Worked Example

Same mei tree dataset used in previous chapter to see if any new discoveries can be made by incorporating the time component and fitting the growth parameters simulatneously throughout the procedure.  


if it doesn't take too much time (maybe by the time defense comes around) cross validate the selection proceudre by leaving out samples and running model again.  How many times do you get the same results?  LOOCV tables

Also consider with holding predictors from the dataset to see if the selection is changed at all.  Start with random ones and then maybe strategically select important predictors to remove to see how it affects the fit


## Discussion

* Downfalls or areas of concern for fitting the genetic effects to the polynomials?
* How do we know there is epistasis?
* how do you keep it from over selecting
* what if it doesn't follow the heredity or hierarchy principles?
* other types of functional/growth equations to represent biologically meaningful scenarios



