# Conclusions

## Summary

<p> In my dissertation I have applied, adapted and extended the forward selection procedure under the marginality assumption first proposed by \cite{hao2014interaction} that is able to reduce the search space substantially by making some reasonable assumptions about the data and underlying biological structure.  These assumptions can be relaxed a little to broaden the scope of the search space if desired. The procedure has been applied in both simulated and real datasets in all areas of study.  There are many advantages to using such a procedure with some being it is a computationally efficient way for high dimensional data situations that arise from high throughput data, especially when considering epistasis between gene markers or other type of interaction effects in the model.  The search space is guided based off of commonly used principles in model selection that are relevant in real world situations. It is able to perform a GWAS with included epistatic interactions a fairly quick manner.  </p>

<p> The ability to relax these assumptions leads to a widened search space but may decrease the speed of the algorithm.  This will make the search more comprehensive and able to pick up on effects that may have been missed with stricter assumptions.  One nice use for the procedure is the ability to screen for complex scenarios that would be hard to find in a lab setting otherwise.  The screening lends itself to find previously undiscovered effects but with the flexibility of the initial model, false positive rates could be a concern and need to be taken into consideration.  Adjustment to the significance level and the model selection criteria have been made to account for this.  This would still need to be followed up by independently checking with other datasets and/or lab verification techniques.  </p>




### HighDeQTL

In [Chapter 02](#highdeqtl) the iFORM procedure was introduced and explored.  It showed many nice properties that are desirable in high dimensional data analysis.   
 - advantages 
 - disadvantages
 steps to improve

### Higher Order Epistasis

In [Chapter 03](#highorder) higher order epistsis was the focus.  Extending the iFORM procedure to open up the possibilty of higher order epistasis is pivotal in understanding complex traits. More support is coming out that phenotypes that are controlled by a multitude of genetic and epigentic effects potential could come from effects involving three or more loci.  

 - advantages 
 - disadvantages
 steps to improve



### iForm Functional Mapping
 - advantages 
 - disadvantages
 steps to improve

In [Chapter 04](#iformfunc)
Functional components were also considered to such a selection procedure.  This also increases the computational complexity of the model and therefore decreases some efficiency gains previously seen.  Using biologically relevant information to guide the fitting of the model.  Initially considers well established growth equations as a baseline for the underlying structure.  It then performs a GWAS level analysis with considering epstatic interactions throughout the process.  


what else will be done before submitting for publication.  

What areas need developed more and what areas will be looked into and discussed in further detail?


## Discussion

**Needs Expanded**

The selection procedures proposed attempt to address problems that are complex and have many moving parts to the procedure.  Using a flexible model is very helpful to fulfill such requirements but this could have it be prone to over fitting at times if not well controlled.  Need to heavily consider this as part of interpreting the model.  Using stricter selection criteria and corrections to multiple testing for significance is extremely important.  The main purpose is to use a screening to guide future research, use for exploratory data analysis and hypothesis generation.  

With the complexity and expense that comes with genetic mapping, especially with a functional traits that needs repeatedly measured over time.  Correlation structures are inherent in this data that need be addressed and usually done so at a heavy computational cost.  Framing the problem as a regression problem and being able to use OLS calculations help to reduce the cost.  Verification steps are important steps especially with something as intricate as epistatic effects between gene markers.

Further investigations are needed to confirm or modify our findings by QTL mapping in natural populations.



**Also included in iFormFunctional Mapping Part**

Monitoring the change in expression patterns over time provides the distinct possibility of unraveling the mechanistic drivers characterizing cellular responses. Gene arrays measuring the level of mRNA expression of thousands of genes simultaneously provide a method of high-throughput data collection necessary for obtaining the scope of data required for understanding the complexities of living organisms. Unraveling the coherent complex structures of transcriptional dynamics is the goal of a large family of computational methods aiming at upgrading the information content of time-course gene expression data. In this review, we summarize the qualitative characteristics of these approaches, discuss the main challenges that this type of complex data present, and, finally, explore the opportunities in the context of developing mechanistic models of cellular response. \cite{androulakis2007analysis}





## Future Steps

 - Aim 1
   - Continue testing properties of the iForm procedure used with the Legendre Polynomials
   - Submit by Summer

 - Aim 2
   - Consider other multivariate responses and correlated error structures.  
   - Expression levels of different cell lines or tissue types as multivariate response 

 - Aim 3
  - Combine functional component and multivariate response into complete functional network
    - Time-Series of gene expression from cell lines or tissue types

More extensive explanation of future aims will be necessary for the final write up.  



### Aim 1

Incorporating other mean curves for the intercept term could help extend and relate to other areas of biology that follow a functional trait.  There are also other types of orthogonal polynomials that could be explored as well.  Using others polynomials would allow for other fits to the data that may be more applicable in other scenarios.  Also using other basis functions in general could open up opportunities for other areas of application.  


## Aim 2
Other interesting areas would be to consider different levels of interactions with other omics data.  Gene-gene interactions considered are only a portion of the picture and this type of modeling could also handle more levels of interactions that occur in a biological system.  One area that I am particularly interested in applying would be in methylation studies of the gene expression.  This level of interaction is very important and a selection procedure like the one proposed could help screen and generate of possible effects and hypothesis to continue to look into.  It would also apply in gene-environment interactions.  This does not have to be restricted to just within the biological system under study.  Environmental factors are important areas that could vastly impact the development of a phenotype.  


### Aim 3

Statistical areas that could as be considered to extend the model would be to include multivariate responses to the system.  For example having gene and protein expression being a bivariate response and to see how genetic markers and epistasis between the markers would better predict this by taking the correlation between those two response variables into account.  Other statistical considerations would be to extend selection criteria to help further reduce the possibility of false positives being selected given a growing dataset like the one that occurs while dynamically including interaction effects throughout the selection.  Making the model even more computational efficient would always be beneficial as well.  The faster a model can accurately run the more likely a researcher will use it.  It will also help process all the high throughput data that is being constantly developed.  

### Closing Remarks

Continuing on, my aim would be to work with datasets of this scale and incorporate the types of statistical methods mentioned and machine learning techniques to aid in analysis. The results could help gain larger insights into the genomic/epigenetic architecture of biological systems. On top of the importance of a functional component to the phenotype, considering other types of multivariate responses would be interesting to study in context of such a system. Integrating different level of omics data and the challenges that arise with such complicated and large datasets has interested me throughout my PhD work. Translating such a complex system into usable information that can be shared in order to prevent and fight disease would be ideal research for me. This type of research would need both methodological development as well as application of existing statistical and machine/deep learning techniques to handle the magnitude of the problem.
