<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="My Dissertation Written in Bookdown.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My Dissertation Written in Bookdown." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="My Dissertation Written in Bookdown." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="iformfunc.html">
<link rel="next" href="appendix.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css, toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

true

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="conclusions-1" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Conclusions</h1>
<div id="summary" class="section level2">
<h2><span class="header-section-number">5.1</span> Summary</h2>
<p>
In my dissertation I have applied, adapted and extended the forward selection procedure under the marginality assumption first proposed by . This procedure is able to reduce the search space substantially by making some reasonable assumptions about the data and underlying biological structure. These assumptions can be relaxed a little to broaden the scope of the search space if desired. The procedure has been applied in both simulated and real datasets in all areas of study. There are many advantages to using such a procedure with some being it is a computationally efficient way for high dimensional data situations that arise from high throughput data, especially when considering epistasis between gene markers or other type of interaction effects in the model. The search space is guided based off of commonly used principles in model selection that are relevant in real world situations. It is able to perform a GWAS with included epistatic interactions a fairly quick manner.
</p>
<p>
The ability to relax these assumptions leads to a widened search space but may decrease the speed of the algorithm. This will make the search more comprehensive and able to pick up on effects that may have been missed with stricter assumptions. One nice use for the procedure is the ability to screen for complex scenarios that would be hard to find in a lab setting otherwise. The screening lends itself to find previously undiscovered effects but with the flexibility of the initial model, false positive rates could be a concern and need to be taken into consideration. Adjustment to the significance level and the model selection criteria have been made to account for this. This would still need to be followed up by independently checking with other datasets and/or lab verification techniques.
</p>
<div id="highdeqtl-1" class="section level3">
<h3><span class="header-section-number">5.1.1</span> HighDeQTL</h3>
<p>
In <a href="intro.html#highdeqtl">Chapter 02</a> the iFORM procedure was introduced and explored. It showed many desirable properties for high dimensional data analysis. Simulation studies were performed to assess the properties and compare across other models. After simulations were conducted a real world application was conducted on gene expression data and conducting a GWAS to find eQTLs for the different genes. Using all markers in a single model as well as the inclusion of the possibility to include epistatic effects as eQTLs gave a novel take on the reanalysis of the C Elegans dataset. As we have seen there are many advantages to the approach but one limiting question presented was that the restriction on epistatic effects may have been too restrictive. Using the strong heredity principle may miss some interactions that could occur between markers that have not yet been selected. Also only considering interactions with markers at two loci instead of higher orders could be limiting the search to fully explain the expression profile.
</p>
</div>
<div id="higher-order-epistasis-1" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Higher Order Epistasis</h3>
<p>
In <a href="highorder.html#highorder">Chapter 03</a> higher order epistasis was focused on the help explain a larger proportion of the phenotypic variation. Extending the iFORM procedure to open up the possibility of higher order epistasis is pivotal in understanding complex traits. More support is coming out that phenotypes that are controlled by a multitude of genetic and epigenetic effects potential could come from effects involving three or more loci (). Given that this is the case the procedure was extended and simulation studies were conducted to see the affect on the properties of the procedure. Different levels of heredity and different orders of interactions were also assessed. Other types of interaction models were also presented for comparison purposes. The iFORM procedure performed well in the simulation studies and appeared to capture the relevant information from the underlying models. A real world application was then presented on a growth parameter of Mei trees. The difference in running the procedure with the possibility of finding higher order epistasis allowed for a much greater proportion the variance to be explained by the markers. This was favorable for the inclusion of higher order effects and showed promise for the model. The phenotype was a static growth parameter, which was the rate at which the trees grew for each individual observation. The growth was measured over time and then a growth curve was fit for each tree. This was then used for the phenotype. This was a good approach but it misses the repeated measures and possibly other information that may have been present throughout the growth process. Since this is the case a functional phenotype was considered to help capture all relevant information.
</p>
</div>
<div id="iform-functional-mapping" class="section level3">
<h3><span class="header-section-number">5.1.3</span> iForm Functional Mapping</h3>
<p>
In <a href="iformfunc.html#iformfunc">Chapter 04</a> functional components were also considered for the selection procedure. This increases the computational complexity of the model and therefore decreases some efficiency gains previously seen but is necessary to capture all information. Using biologically relevant information to guide the fitting of the model to a functional curve will help improve the efficiency and relevance. Initially well established growth equations were considered as a baseline for the underlying structure. The procedure then performs a GWAS level analysis with considering epistatic interactions throughout the process. The selection of the markers as possible genetic and epigenetic effects follows in a similar fashion as the iForm procedure. Additional aspects had to be accounted for the functional response and growth curve. Again simulation studies were performed and a real world application was conducted. A reanalysis of the full Mei tree dataset used in chapter 3 was explored. Including the functional valued phenotype gave new insight into the genetic control.
</p>
</div>
</div>
<div id="discussion-3" class="section level2">
<h2><span class="header-section-number">5.2</span> Discussion</h2>
<p>
The selection procedures proposed attempt to address problems that are complex and have many moving parts to the procedure. Using a flexible model is very helpful to fulfill such requirements but this could lend itself to be prone to over fitting at times if not well controlled. Being aware of this factor is part of interpreting the model a researcher would need to be aware of. Using stricter selection criteria and corrections to multiple testing for significance is extremely important. The main purpose is to use as a screening to guide future research while considering the possibilities of epistatic effects and accounting for the entire genetic architecture at one time. The importance of such a procedure could be used to help guide research and also find new discoveries into complex, multi-loci traits that would be extremely difficult to find in just a lab setting alone.
</p>
<p>
There are many attractive properties that the iFORM procedure posses. It is computationally efficient given the large and growing predictor space it uses. Along with the computational advantages, the selection procedures proposed have sure screening properties. This indicates that all the important effects in the true model will be selected with probability tending to one. It also only includes a small proportion of effects based off of a designated criteria that can be set. This will help with interpretability of the final output as well as controlling for the possibility of overfitting the data. There is still much that can be extended and explored by using this type of selection procedure on a variety of settings and/or phenotypes.
</p>
</div>
<div id="future-steps" class="section level2">
<h2><span class="header-section-number">5.3</span> Future Steps</h2>
<div id="aim-1" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Aim 1</h3>
<p>
Incorporating other curves for the mean response curve could help extend and relate to other areas of biology that follow a functional trait. There are also other types of orthogonal polynomials that could be explored as well. Using others polynomials would allow for other fits to the data that may be more applicable in other scenarios. Also using other basis functions in general could open up opportunities for other areas of application. There are many non-linear functions that could be used besides a polynomial fit that may also explain the underlying genetic effects over time for other situations.
</p>
</div>
</div>
<div id="aim-2" class="section level2">
<h2><span class="header-section-number">5.4</span> Aim 2</h2>
<p>
Other interesting areas would be to consider different levels of interactions with other omics data. The gene-gene interactions considered are only a portion of the picture and this type of modeling could also handle complex levels of interactions that occur in a biological system. One area that I am particularly interested in applying it to would be in methylation and its interaction with gene expression. This level of interaction is very important and a selection procedure like the one proposed could help screen and generate possible effects for hypotheses to continue to look into. This does not have to be restricted to just within the biological system under study. It would also apply in gene-environment interactions. Environmental factors are important areas that could vastly impact the development of a phenotype. Having the interaction with environmental factors could open up further avenues that able to previously be explored.
</p>
<div id="aim-3" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Aim 3</h3>
<p>
Statistical areas that could as be considered to extend the model would be to include multivariate responses to the system. For example having gene and protein expression being a bivariate response and to see how genetic markers and epistasis between the markers would better predict this scenario. It could take the correlation between the protein and expression response variables into account. Other statistical considerations would be to extend selection criteria to help further reduce the possibility of false positives being selected given a growing dataset like the one that occurs while dynamically including interaction effects throughout the selection. Making the model even more computational efficient would always be beneficial as well. The faster a model can accurately run the more likely a researcher will be to use it. It will also help process all the high throughput data that is being constantly developed.
</p>
</div>
<div id="closing-remarks" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Closing Remarks</h3>
<p>
Continuing on, my aim would be to work with datasets of high dimensional scale and incorporate the types of statistical methods mentioned and machine learning techniques to aid in analysis. The results could help gain larger insights into the genomic/epigenetic architecture of biological systems. On top of the importance of a functional component to the phenotype, considering other types of multivariate responses would be interesting to study in context of such a system. Integrating different level of omics data and the challenges that arise with such complicated and large datasets has interested me throughout my work. Translating such a complex system into usable information that can be shared in order to prevent and fight disease would be ideal research. This type of research would need both methodological development as well as application of existing statistical and machine/deep learning techniques to handle the magnitude of the problem.
</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="iformfunc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-Conclusions.Rmd",
"text": "Edit"
},
"download": ["Defense.pdf", "Defense.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
