<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="My Dissertation Written in Bookdown.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My Dissertation Written in Bookdown." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="My Dissertation Written in Bookdown." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="iform-functional-mapping-a-computational-method.html">
<link rel="next" href="appendix.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css, toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

true

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="conclusions" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Conclusions</h1>
<div id="summary" class="section level2">
<h2><span class="header-section-number">5.1</span> Summary</h2>
<p>In my disseration I have applied, adapted and extended the forward selection procedure under the marginality assumption first proposed by  that is able to reduce search space substantially by making some reasonable assumptions about the data. These assumptions can be relaxed a little to broaden the scope of the space. The procedure ahs been applied in both simulated and real datasets in all areas of study. There are many advantages to using such a procedure with some being it is a computationally efficient way for high dimensional data situations that arise from high throughput data, especially when considering epistasis between gene markers or other type of interaction effects in the model. Guided search space based off of commonly used principles in model selection that are relevant in real world situations. It is able to perform a GWAS with included epistatic interactions a fairly quick manner. Able to relax assumptions to widen the search space. This may decrease the speed of the algorithm but makes the search more comprehensive. Also it is able to screen for complex scenarios that would be hard to find in a lab setting otherwise. Having said that with the flexiblity of the initial model, false posiive rates could be a concern and need to be taken into consideration. Adjustement to the significance level and the model selection criteria have been made to account for this. This would still need to be followed up on independtly by checking with other datasets and/or lab verification.</p>
<p>Functional componenets were also considered to such a selection procedure. This also increases the computational complexity of the model and therefore decreases some efficeiency gains previously seen. Use biologically relevant information to guide the fitting of the model. Initially considers well established growth equations as a baseline for the underlying structure. It then performs a GWAS level analysis with considering epstatic interactions throughout the process.</p>
</div>
<div id="discussion-3" class="section level2">
<h2><span class="header-section-number">5.2</span> Discussion</h2>
<p>The selection procedures proposed attempt to address problems that are complex and have many moving parts to the procedure. Using a flexible model is very helpful to fulfill such requirements but this could have it be prone to over fitting at times if not well controlled. Need to heavily consider this as part of interpretating the model. Using stricter selection criteria and corrections to multuple testing for significance is extremely important. The main purpose is to use a screening to guide future research, use for exploratory data analysis and hypothesis genration.</p>
<p>With the complexity and expense that comes with genetic mapping, especially with a functional traits that needs repeatedly measured over time. Correlation structures are inherent in this data that need be addressed and usually done so at a heavy computational cost. Framing the problem as a regression problem and beingn able to use OLS calculations help to reduce the cost. Verification steps are important steps especially with something as intricate as epistatic effects between gene markers.</p>
<p>Further investigations are needed to confirm or modify our findings by QTL mapping in natural populations.</p>
</div>
<div id="future-steps" class="section level2">
<h2><span class="header-section-number">5.3</span> Future Steps</h2>
<p>More extensive explaination of future aims will be necessary for the final write up.</p>
<div id="aim-1" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Aim 1</h3>
<p>Incorporating other mean curves for the intercept term could help extend and relate to other areas of biology that follow a functional trait. There are also other types of orthogonal polynomials that could be explored as well. Using others polynomials would allow for other fits to the data that may be more applicable in other scenarios. Also using other basis functions in general could open up opportunties for other areas of application.</p>
</div>
</div>
<div id="aim-2" class="section level2">
<h2><span class="header-section-number">5.4</span> Aim 2</h2>
<p>Other interesting areas would be to consider different levels of interactions with other omics data. Gene-gene interactions considered are only a portion of the picture and this type of modeling could also handle more levels of interactions that occur in a biological system. One area that I am particularly interested in applying would be in methylation studies of the gene expression. This level of interaction is very important and a selection procedure like the one proposed could help screen and generate of possible effects and hypothesis to continue to look into. It would also apply in gene-environment interactions. This does not have to be restricted to just within the biological system under study. Environmental factors are important areas that could vastly impact the development of a phenotype.</p>
<div id="aim-3" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Aim 3</h3>
<p>Statistical areas that could as be considered to extend the model would be to include multivariate reponses to the system. For example having gene and protein expression being a bivariate reponse and to see how genetic markers and epistasis between the markers would better predict this by taking the correlation between those two response variables into account. Other statistical considerations would be to extend selection criteria to help further reduce the possibility of false positves being selected given a growing dataset like the one that occurs while dynamically including interaction effects throughout the selection. Making the model even more computational efficient would always be benefical as well. The faster a model can accurately run the more likely a researcher will use it. It will also help process all the high throughput data that is being constantly developed.</p>
</div>
<div id="closing-remarks" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Closing Remarks</h3>
<p>Continuing on, my aim would be to work with datasets of this scale and incorporate the types of statistical methods mentioned and machine learning techniques to aid in analysis. The results could help gain larger insights into the genomic/epigenetic architecture of biological systems. On top of the importance of a functional component to the phenotype, considering other types of multivariate responses would be interesting to study in context of such a system. Integrating different level of omics data and the challenges that arise with such complicated and large datasets has interested me throughout my PhD work. Translating such a complex system into usable information that can be shared in order to prevent and fight disease would be ideal research for me. This type of research would need both methodological development as well as application of existing statistical and machine/deep learning techniques to handle the magnitude of the problem.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="iform-functional-mapping-a-computational-method.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-Conclusions.Rmd",
"text": "Edit"
},
"download": ["Defense.pdf", "Defense.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
