<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="My Dissertation Written in Bookdown.">
  <meta name="generator" content="bookdown 0.3.8 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My Dissertation Written in Bookdown." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="My Dissertation Written in Bookdown." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="highdeqtl.html">
<link rel="next" href="iform-functional-mapping-a-computational-method.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css, toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

true

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="highorder" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> High-order Epistatic Networks</h1>
<div id="motivation-1" class="section level2">
<h2><span class="header-section-number">3.1</span> Motivation</h2>
<strong>Start of My Paper</strong>
<p>
Quantitative traits are very difficult to study because these traits are controlled by many genes that interact in a complicated way (; ). Genome-wide mapping and association studies increasingly available due to next-generation high-throughput genotyping techniques have proven to be useful for characterizing gene-gene interactions, coined epistasis, that contribute to phenotypic variation (; ; ). Powerful statistical methods have been developed to analyze all possible markers simultaneously, from which to search for a complete set of epistasis for quantitative traits (; ). The joint analysis of all markers is particularly needed to chart an overall picture of genetic interactions, in comparison with computationally less expensive marginal analysis.
</p>
<p>
Epistasis reported in the current literature is mostly due to interactions between two genes. However, a growing body of evidence shows that genetic interactions involving more than two loci play a pivotal role in regulating the genetic variation of traits (; ; ; ). For example, in a mapping population deriving from crossing two chicken lines, three-locus interactions were detected to determine body weight (). A mapping study established by two yeast strains identified genetic interactions involving five or more loci for colony morphology (). Other studies have demonstrated that high-order epistasis is of critical importance in regulating metabolic networks in yeast () and Escherichia coli and Saccharomyces cerevisiae (; ), whereas lower-order (pairwise) epistasis may be insufficient to explain metabolic variation for these organisms.
</p>
<p>
The theoretical models of high-order epistasis have well been established by mathematical biologists (; ). These models provided a foundation to interpret high-order epistasis from a biological standpoint. A few statistical models have been derived to estimate and test high-order epistasis in case-control designs () and population-based mapping settings ().  developed a Bayesian version of detecting high-order interactions for both continuous and discrete phenotypes. However, these models were based on a marginal analysis, thus less powerful to illustrate a global view of genetic control mechanisms due to high-order epistasis.
</p>
<p>
In this article, we deploy a variable selection procedure within a genetic mapping or association setting to characterize the genetic architecture of complex traits composed of main effects of individual genes, pairwise epistasis between two genes, and three-way epistasis among three genes. The model was built on greedy interaction screening forward selection developed under the marginality principle (named iFORM) by . These approaches, proved to possess sure screening property for ultrahigh-dimensional modeling, have been implemented to model the genetic architecture of main effects and pairwise epistasis due to eQTLs for gene transcripts (). Here, we extend the implementation of iFORM to systematically capture three-way interactions that are expressed among all possible markers studied. To show the statistical power of the extended model, we performed computer simulation studies. The model was further validated through analyzing a real data of genetic mapping for shoot growth in a woody plant, mei (Prunus mume). The model should be used in any other mapping or association studies of quantitative traits.
</p>
</div>
<div id="methods-1" class="section level2">
<h2><span class="header-section-number">3.2</span> Methods</h2>
<div id="mapping-and-association-studies" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Mapping and association studies</h3>
<p>Genetic mapping and association studies are two types of designs used to dissect quantitative traits. The former is based on a controlled cross derived from distinct parents, whereas the latter samples different genotypes from a pool of accessions or a natural population. In both types of design, a set of individuals are sampled to be phenotyped for quantitative traits of interest and genotyped by molecular markers distributed throughout the entire genome. For a particular genetic experiment, the number of markers is much larger than that of samples, thus, it is impossible to estimate the genetic effects of all markers simultaneously using traditional regression models. This issue becomes much intractable when we aim to estimate genetic interactions of different orders. To tackle the issue of the number of predictors &gt;&gt; the number of samples, several variable selection approaches have been implemented in association studies. One approach is forward selection which was shown to be robust for estimating pairwise interactions of predictors (Hao and Zhang 2014). With sure screening properties and controlling for false positives, this approach, named iFORM, performs very well in capturing important information in explaining the response variable. On top of these nice theoretical properties it is computationally efficient by using ordinary least squares calculations and only requiring a predetermined set up steps. Here, we extended the iForm procedure to include HGI’s to capture more relevant information. In the following sections, the notation and model set-up will be introduced. After this theoretical properties will be explored. Finally simulated and real data analysis will be conducted to help confirm the theoretical properties and show the feasibility of using the model for screening across whole genomes to more precisely explain phenotypes of interest.</p>
</div>
<div id="epistatic-model" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Epistatic model</h3>
<p>Consider a linear model that underlies the true genotype-phenotype relationship. Assume that the phenotype, as the response of the model, is controlled by a set of p SNPs that act singly and/or interact with each other. These main and interaction effects of markers, i.e., the predictors of the model, need to be estimated. Let <span class="math inline">\(Y = (y1, …, yn)^T\)</span> denote the phenotypic value of n samples from a mapping or association population. When considering pairwise and three-way interactions, the linear model is expressed as</p>
<span class="math display" id="eq:lin-mod">\[\begin{equation}
Y = \alpha + X^T\beta + Z^T\gamma + W^T\eta + \epsilon
\tag{3.1}
\end{equation}\]</span>
<p>where <span class="math inline">\(X=(X_1,…,X_p )^T\)</span> is the design matrix that specifies the genetic effects of each <span class="math inline">\(marker  = (1, …, p) , Z=(X_j X_k )^T (1≤j≤k≤p)\)</span> is the design matrix that specifies the epistatic effects between two markers, expressed in <span class="math inline">\(\gamma, W=(X_j X_k X_l )^T (1 ≤ j ≤ k ≤ l ≤ p)\)</span> is the design matrix that specifies the epistatic effects among three markers, expressed in <span class="math inline">\(η\)</span>, and <span class="math inline">\(\epsilon ~ N(0,\sigma^2 )\)</span> is the residual error normally distributed with mean zero and variance σ^2. We denote the index sets for the linear, order-2 and order-3 effects in equation (1), respectively, as <span class="math inline">\(P_1={1,2,…,p}\)</span> <span class="math inline">\(P_2={(j,k):1≤j≤k≤p}\)</span> <span class="math inline">\(P_3={(j,k,l):1≤j≤k≤l≤p}\)</span> With the significant main, order-2 interaction and order-3 interaction effect sets being, <span class="math inline">\(T_1={j:β_j≠0,j∈P_1 }\)</span> <span class="math inline">\(T_2={(j,k):γ_jk≠0,(j,k)∈P_2 }\)</span> <span class="math inline">\(T_3={(j,k,l):η_jkl≠0,(j,k,l)∈P_3 }\)</span></p>
<p>The true size of <span class="math inline">\(T_1\)</span>, will be <span class="math inline">\(p_1\)</span> and similarly for <span class="math inline">\(T_2\)</span> and <span class="math inline">\(T_3\)</span> will have sizes <span class="math inline">\(p_2\)</span> and <span class="math inline">\(p_3\)</span> respectively. There will be a total of 3 sets referred to throughout the procedure, the candidate set <span class="math inline">\(C\)</span>, the selection set <span class="math inline">\(S\)</span> and the model set, <span class="math inline">\(M\)</span>. The candidate set is the set of all possible predictors at a given step in the selection process. The selection set contains the predictors that have previously been selected from the candidate set from each iteration of the procedure. Finally, the model set is the final model that is fit from the selection set at the end of the procedure. The BIC is used to determine the optimal cutoff for the final model size.</p>
</div>
<div id="iform-with-high-order-epistasis" class="section level3">
<h3><span class="header-section-number">3.2.3</span> iForm with high-order epistasis</h3>
<p>The iForm procedure is a forward selecting procedure. In traditional forward selection the procedure starts with the empty set and then iterates through the entire set of possible predictors in <span class="math inline">\(C\)</span> and selects the best predictor and includes it in S at the end of each step. The best predictor can be determined in many ways but usually is defined by the predictor that results in the least amount of error. For our purposes we use the residual sum of squares. This continues with selecting the best predictor from C at each step until a designated stopping criterion is met or until some information criterion is met. Common information criteria used for selecting predictors to be in <span class="math inline">\(M\)</span> are AIC, BIC, <span class="math inline">\(R^2\)</span> and Mallow’s <span class="math inline">\(C_p\)</span> statistic.</p>
<p>The iForm procedure for high-order epistatic detection parallels the forward selection procedure, but C will grow dynamically with the creation of order-2 and order-3 interaction effects between main effects that were included from previous iterations of the procedure. There are three steps to the model selection. The first step is to initialize the 3 sets mentioned above. The sets, <span class="math inline">\(S\)</span> and <span class="math inline">\(M\)</span> are set to the empty set while the candidate set, <span class="math inline">\(C\)</span>, is first set to <span class="math inline">\(P_1\)</span>, all the main effects. The next step starts the forward selection procedure selecting predictors from <span class="math inline">\(C\)</span>. The selected predictor will be a main effect at the first step. At subsequent steps, after interaction effects are included, selected predictors could be either be a main effect, order-two or order-three interaction effect. The final step involves repeating the second step until a designated stopping criterion is met. This can be a certain amount of predictors to be considered in the final model, or it can be based off of other factors such as the sample size. The designated stopping criterion will be denoted as d. For our purposes we use d as a function of the sample size, <span class="math inline">\(d = n/lo g_2a(n)\)</span>. The procedure will run up until d iterations, and the optimal model will then be constructed from the selection set. This is done by an information criterion. Here we used the Bayesian Information Criterion proposed by  denoted as the <span class="math inline">\(BIC_2\)</span>. This was derived by them to control the false discovery rate in high dimensional model selections.</p>
<span class="math display" id="eq:bic2">\[\begin{equation}
BIC_2(\hat{M}) = log(\hat{σ_M}^2) + n^{-1} |\hat{M}| * (log(n) + 2*loga(d^*))
\tag{3.2}
\end{equation}\]</span>
<p>Once the selection procedure is done and there are d predictors in the selection set the BIC is used to determine the cutoff value for the optimum number of predictors in the model set. Then linear regression is performed on the model set.</p>
<p>Two guiding principles are used to help dynamically select the main effects and epistasis effects throughout the procedure. The first is the marginality principle, which states that an effect will not be removed from the model once it has been selected. A previous selected effect may become marginal by the inclusion of subsequent effects. This especially can be the case when an interaction effect is included. One of the parent effects may become less significant or even not significant at all by considering both in the model. The next principle we state as the heredity principle but has also been referred to in other work as the hierarchy principle (Bien et al 2013 and Lim and Hastie 2014).</p>
<p>The heredity (hierarchy) principle help reduce the search space by making the assumption that previously selected main effects would be involved in the interaction effects. By considering this principle it substantially reduces the search space making this feasible for ultra-high dimensional situations. Even larger than ram datasets can be used with efficient memory mapping of the dataset while running the procedure. The weak version of the heredity principle for three-way interactions states that at least one of the main effects needs to be selected into the model to consider an interaction effect that contains that predictor. Considering a moderately high set of predictors say p = 5000, if trying to include all order-2 interactions upfront, will make the candidate set be as high as 12,498,000. This alone could exceed most ram requirements of standard computers. This is before even stepping up to order-3 interactions. The weak heredity principle would decrease the candidate set substantially. Assuming a sample size of <span class="math inline">\(n = 200\)</span>, would give a cut off of <span class="math inline">\(n/log_{2}(n) = 200/log_{2}(200) = 26\)</span> steps in the procedure. This would give a maximum of approximately 135,000 candidate predictors. This gives a 100 fold decrease in the candidate set. This could substantially make ultra-high dimensional analysis more feasible and also speed it up in the process. This is the weak case. If considering the strong case the decrease in candidate space is even more apparent. Aside from the efficiency by lowering the search space of the candidate set, the heredity principle is usually taken into account by researchers when selecting models involving the consideration for interaction effects.</p>
</div>
<div id="theoretical-properties" class="section level3">
<h3><span class="header-section-number">3.2.4</span> Theoretical Properties</h3>
<p>The theoretical properties of the iForm procedure with high-order epistasis follow closely with the forward selection procedure.  summarize forward selection nicely as follows. At each step, the response is regressed on the most correlated covariate, and the residual is calculated and used as the new response in next step. After the most correlated covariate (say, <span class="math inline">\(X_1\)</span>) is selected, all other covariates are regressed on <span class="math inline">\(X_1\)</span>, and then the covariates are substituted by the corresponding normalized residuals, which are used as the new covariates in next step. By viewing forward selection in this sense the computational complexity of the procedure depends upon the size of the candidate set. The candidate set in the iForm’s case does grow dynamically at each step, by at most the number of predictors currently selected in C for each step. If we denote the current size of the candidate set as m then each iteration of the procedure grows with complexity of O(nm), where n is the sample size. Leaving the selection unrestricted we would not be able to fit more than n predictors for a linear model and therefore n would be the most main effects that would be able to be selected. Considering the weakest form of the heredity principle at the current iteration there would be at most <span class="math inline">\(p + (n(n-1)(n-2))/6\)</span> predictors in the candidate set. This would make the total complexity of the selection procedure to be <span class="math inline">\(nO(n(p+n(n-1)(n-2))) = O(n^3 p+n^5)\)</span>. This makes the total complexity grow linearly as p grows.</p>
<p>The theoretical properties of the iForm procedure show sure screening properties (). By this we mean that all the import predictors, whether that is a main effect or epistatic effect will be selected with probability tending to 1. This is important to capture as much of the signal as possible through all the noise that comes with <span class="math inline">\(p &gt;&gt; n\)</span> or ultra-high dimensional situations. It is also important not to ‘over-fit’ the model with unnecessary predictors that actually explain more noise in the data that the model is being fitted on than the actual signal you would like to pick up on.</p>
<p>To show the property from above the following conditions would need to be met for regulatory purposes. Hao and Zhang (2014) showed how under these conditions sure screening properties for interaction models like FS2 and iForm are satisfied. This also applies to order-3 interaction models like FS3 and iForm with higher order epistasis, like we do with the high-order epistasis model. The following assumptions need to be met for these conditions. The first is that the <span class="math inline">\(X=(X_1,…,X_p )^T\)</span> are jointly and marginally normal with independent normally distributed error. Next we would need the eigenvalues of the covariance matrix to be positive and bounded by two constants <span class="math inline">\(0 &lt; \tau_min &lt; 1 &lt; \tau_{max} &lt; \infty\)</span>, such that <span class="math inline">\(\sqrt{\tau}_min &lt; \lambda_min (\Sigma)≤\lambda_max (\Sigma)&lt; \sqrt{\tau}_max/4\)</span>. Also, the genetic effects, <span class="math inline">\(\beta\)</span> need a certain level of signal strength. This we would assume to be <span class="math inline">\(|(|\beta|)| ≤ C_\beta\)</span> for some positive constant <span class="math inline">\(C_\beta\)</span> and <span class="math inline">\(\beta_min ≥ \nu\beta \eta^{-\xi_min}\)</span>,with <span class="math inline">\(\beta_min=mina(\beta)\)</span>. Lastly, there needs to be a certain level of sparsity to the number of important effects. Denoting the total number of important effects as <span class="math inline">\(d_0\)</span>, and positive constants <span class="math inline">\(\xi,\xi_0\)</span> and <span class="math inline">\(\nu\)</span> we would need <span class="math inline">\(log(p) ≤ \nu n^\xi,d_0 ≤ \nu n^(\xi_0)\)</span> and <span class="math inline">\(ξ+6ξ_0+12ξ_min&lt;1/2\)</span>. The conditions stated are accepted standards in the literature when studying ultra-high dimensional situations. ( , ; ).</p>
</div>
</div>
<div id="application-1" class="section level2">
<h2><span class="header-section-number">3.3</span> Application</h2>
<div id="simulation-studies" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Simulation Studies</h3>
<p>To study the numeric properties of the selection procedure, simulation studies were conducted. To capture relevant data structures, there were several different scenarios considered. For each scenario 50 predictors were generated with a sample size of 300 observations. The data was split into training and a testing set to study both the fitted properties of the model as well as the generalizability of the model. There were a variety of metrics obtained to assess the suitability of each model utilized in the simulations. The first metrics that were taken into account were the rates for the true positives, false positives, true negatives and false negatives. Since we have a variety of levels to each of the models each of the rates were evaluated for the different hierarchical levels. Some of the models only have main effects and/or two-way interactions, therefore the rates were only given for the area applicable to model and the rest were reported as NA. The generalizability of the models was also assessed by withholding 100 random observations as a test set. All the data was generated from the same scenario and then 100 of the observations were randomly selected and stored for out of sample measures. The data was generated from the given scenario and randomly split before assessing the models. The exact same training and testing sets were used to fit and assess each of the models in order to make as fair of a comparison as possible. Each scenario was replicated 100 times and measures were averaged over all replicates. The two measures assessed were mean square error and the coefficient of determination. The analogous in-sample measures were also calculated for comparison. The models being compared in the simulation studies are Forward Selection, Forward Selection with all order-2 interactions (FS2), Forward Selection with all order-2 interactions (FS3), iForm strong heredity order-2, iForm weak heredity order-2, iForm stong heredity order-3, iForm weak heredity order-3, Glinternet (), and finally hierNet ()</p>
<p>Covering a variety of settings the following scenarios were evaluated and compared. The first is where the data were generated from the interactions of the model follow a strong heredity (hierarchy) with <span class="math inline">\(\sigma = 1\)</span>. The second, the data is generated to have the interactions in follow a weak heredity (hierarchy) with <span class="math inline">\(\sigma = 1\)</span>. The third scenario is anti-heredity (hierarchical) where the interaction effects are only among predictors not present as main effects in the model. Finally the last scenario on generates data that come from pure interactions between predictors with no main effects present in the model used to generate the data.</p>
<p>For the first scenarios where the truth obeys strong heredity where all of the parent main effects need to be selected before interactions are selected. The models that appeared to do the best in this simulation were forward selection on all order-3 interactions included from the beginning (FS3), iForm order-3 weak heredity and iForm order-3 strong heredity (Table 1). The FS3 took over a 40 fold increase in time to run. The other comparison models, glinternet and hierNet seemed to perform well on the training set but not as well on the testing set. This would indicate that some overfitting was occurring with those types of regularization models. The next scenario was when the truth obeys weak heredity. With the underlying model obeying the weak heredity, the iForm order-3 strong heredity version dropped off in performance slightly. However, the FS3 and iForm order-3 remained as top performers (Table 2). The third scenario assessed was from an underlying model with an anti-heredity structure. Both main effects and interaction effects were used in the model to generate the data. However the interactions included in the model were of combinations of main effects in the candidate set, that were not in the model. The iForm seems to drop in performance with this scenario (Table 3). This is to be expected because it is in direct violation of the underlying assumptions of the model hierarchy. Even with these violations of the heredity it still performed reasonably well. Lastly, making the scenario a little more extreme, the underlying model generating the data was only of interactions. There were no main effects included in the model. The results of this scenario are shown in Table 4. Performance appeared to drop off for all models explored in the simulation.</p>
<p>In the scenarios where the data was assumed to follow some form of a hierarchical structure for the epistasis effects the iForm procedure for higher-order epistasis effects appeared to perform the best. Not only did it result in selecting the correct model, the false positive rate was also among the lowest. The out of sample error was also among the lowest between each of the models compared. With the procedure using OLS calculations, it also performed the fastest out of the models including epistasis effects. All of the combined show the promise of the iForm procedure for GWAS type studies. With the other scenarios, the underlying structure of the data does not follow a typical intuition about the structure of data in biology.</p>
</div>
<div id="worked-example" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Worked Example</h3>
<p>We validated the biological usefulness of the model by analyzing a mapping data for a woody plant, mei (Prunus mume). Originated in China, mei has been cultivated for its ornamental flowers for thousands of years (, ). Its many desirable properties, such as cold-hardiness, colors and flavors, are appraised as a symbol of persistence and beauty in Chinese culture. Recent sequencing of its genome has made it an ideal model system to study the genetics and evolution of woody plants (). To improve the growth rigor and form of mei important to its ornamental value, a cross was made between two distinct cultivars, Fenban (female parent) and Kouzi Yudie (male parent), aimed to select superior genotypes from hybrids. To the end, an F1 mapping population of 190 hybrids was established and further genotyped for 4,934 SNP markers over eight mei chromosomes.</p>
<p>To test genotypic differences in growth performance, each of these hybrids was grafted on an established root stock using multiple budding scions. Next spring, buds on the scions sprouted into shoots. The lengths and diameters of 10 randomly selected shoots were measured once every two weeks during an entire growth season from March to October. It was found that both shoot length and diameter growth was well fitted to the three-parameter growth equation expressed as</p>
<span class="math display" id="eq:growth-curve">\[\begin{equation}
g(t) = a/[1+b*exp(-rt)]
\tag{3.3}
\end{equation}\]</span>
<p>where <span class="math inline">\(g(t)\)</span> is the amount of shoot growth at time t, a is the asymptotic value of growth when time tends to be infinite, b is a parameter that reflects the amount of growth at time 0, and r is the relative growth rate. These three parameters determine the overall form of growth curve jointly, although they function differently. Thus, by estimating these parameters for individual hybrids using a nonlinear least squares approach, we can draw the growth curve of each hybrid. Differences in growth curve among hybrids may be controlled by specific genes or quantitative trait loci (QTLs). Although tremendous efforts have been made to map growth QTLs and their epistasis (;; ), none has characterized the contribution of high-order epistasis although it has been thought to regulate growth processes.</p>
<p>By treating the estimates of growth parameters for individual hybrids as “phenotyopic traits,” we used iFORM to map growth QTLs and QTL-QTL interactions. Of 4,934 markers, 00 are the testcross markers at which markers are segregating due to only one heterozygous parent and 00 are the intercross markers whose segregation results from the heterozygosity of both parents. For a testcross marker, there is only one main genetic effect, whereas an intercross marker contains additive and dominant main effects. Thus, a pair of testcross markers produces only type of epistasis, but a pair of intercross markers forms four types of epistasis, additive × additive, additive × dominant, dominant × additive and dominant × dominant. For two markers with one from the testcross and the other from the intercross, there are two types of epistasis, i.e., additive × additive and additive × dominant (). The number and type of epistasis can be characterized for any three markers accordingly. Here, the iFORM was implemented in a way that allows both marker markers to be modeled and analyzed simultaneously.</p>
<p>To demonstrate the possible importance of high-order epistasis, we analyze the data by assuming that growth parameters are controlled by low-order epistasis only and by both low- and high-order epistasis, respectively. The weak heredity (hierarchical) was used to screen every SNP and possible interaction of the main effects selected and the rest of the SNPs left in the candidate set. It was not restricted to the strong case where both main effects had to be in the model for the interaction to be considered. For the pairwise epistatic model, this grew the candidate set to almost 20,000 predictors to choose from. It turned out that 5 predictors were chosen, i.e., four main additive effects of markers, AATTC_nn_np_2517, AATTC_nn_np_2815, CATG_nn_np_3479 and CATG_nn_np_1284 and one epistatic effect due to markers AATTC_nn_np_2815 and AATTC_lm_ll_3034, for growth parameter r of shoot length (Table 4). The main effect of marker AATTC_lm_ll_3034 was detected to be insignificant. These main and epistatic effects together explained 32.41% of the total variance of parameter r.</p>
<p>When opening up the iForm procedure to the possibility to creating higher order interactions to be placed into the candidate set, a more complete picture of the phenotypical variation was revealed. The amount of predictors included in the final model grew to 12, with one of them being three-way interactions among markers AATTC_nn_np_2815, AATTC_lm_ll_3034 and AATTC_nn_np_1615.The adjusted R2 jumped up to over 70% (Table 4). This astonishing jump in predictive power is an exemplar case as to the importance of higher-order interactions in genetic models. Not only did higher-order interactions become one of the most significant predictors in the model selected, it also allowed for other order-two interactions and main effects to be kept in the model that were previously left out. At the next step of every iteration, the new candidate effect was conditioned on everything previously selected. With the conditional effect of the higher-order interaction it enabled for other lost effects to be modeled as well.</p>
<p>The purpose of the mei genetic project is to study the genetic control of shoot growth form. Here, we further analyze how three-way interactions detected by our model affect growth form. Assume that there are three testcross markers, A (with two alleles A, a), B (with two alleles B, b), and C (with two alleles C, c), which interact jointly to affect shoot growth. The three markers form eight genotypes AABBCC, AABBCc, AABbCC, AABbCc, AaBBCC, AaBBCc, AaBbCC and AaBbCc whose genotypic means at time t are partitioned into different components, respectively, expressed as</p>
<div class="figure" style="text-align: center"><span id="fig:growth-compare"></span>
<img src="images/GrowthCurveComparison.png" alt="Growth Curve Comparison" width="80%" />
<p class="caption">
FIGURE 3.1: Growth Curve Comparison
</p>
</div>
<span class="math display" id="eq:genetic-effects">\[\begin{align}
\mu_{111}(t) &amp;= \mu(t) + \alpha_1(t) + \alpha_2(t) + \alpha_3(t) + i_{12}(t) + i_{13}(t) + i_{23}(t) + i_{123}(t)  \notag \\
\mu_{112}(t) &amp;= \mu(t) + \alpha_1(t) + \alpha_2(t) – \alpha_3(t) + i_{12}(t) – i_{13}(t) – i_{23}(t) – i_{123}(t)  \notag \\
\mu_{121}(t) &amp;= \mu(t) + \alpha_1(t) – \alpha_2(t) + \alpha_3(t) – i_{12}(t) + i_{13}(t) – i_{23}(t) – i_{123}(t)  \notag \\
\mu_{122}(t) &amp;= \mu(t) + \alpha_1(t) – \alpha_2(t) – \alpha_3(t) – i_{12}(t) – i_{13}(t) + i_{23}(t) + i_{123}(t)  \notag \\
\mu_{211}(t) &amp;= \mu(t) – \alpha_1(t) + \alpha_2(t) + \alpha_3(t) – i_{12}(t) – i_{13}(t) + i_{23}(t) – i_{123}(t)  \notag \\
\mu_{212}(t) &amp;= \mu(t) – \alpha_1(t) + \alpha_2(t) – \alpha_3(t) – i_{12}(t) + i_{13}(t) – i_{23}(t) + i_{123}(t)  \notag \\
\mu_{221}(t) &amp;= \mu(t) – \alpha_1(t) – \alpha_2(t) + \alpha_3(t) + i_{12}(t) – i_{13}(t) – i_{23}(t) + i_{123}(t)  \notag \\
\mu_{222}(t) &amp;= \mu(t) – \alpha_1(t) – \alpha_2(t) – \alpha_3(t) + i_{12}(t) + i_{13}(t) + i_{23}(t) – i_{123}(t)
\tag{3.4}
\end{align}\]</span>
<p>where <span class="math inline">\(\mu(t)\)</span> is the population mean at time t; <span class="math inline">\(\alpha_1(t)\)</span>, <span class="math inline">\(\alpha_2(t)\)</span>, and <span class="math inline">\(\alpha_3(t)\)</span> are the genetic effects of markers A, B and C at time t, respectively; <span class="math inline">\(i_{12}(t)\)</span>, <span class="math inline">\(i_{13}(t)\)</span>, and <span class="math inline">\(i_{23}(t)\)</span> are the pairwise epistatic effects between markers A and B, A and C and B and C at time t, respectively; and <span class="math inline">\(i_{123}(t)\)</span> is the three-way epistatic effect among three the markers at time t. From the above equations, we solve the pairwise and three-way epistatic effects as</p>
<div class="figure" style="text-align: center"><span id="fig:epistasis-compare"></span>
<img src="images/EpistasisComparison.png" alt="Epistasis Comparison" width="80%" />
<p class="caption">
FIGURE 3.2: Epistasis Comparison
</p>
</div>
<span class="math display" id="eq:epi-effects">\[\begin{align}
i_{12}(t) &amp;=  [(\mu_{111}(t) + \mu_{112}(t) + \mu_{221}(t) + \mu_{222}(t)) – (\mu_{121}(t) + \mu_{122}(t) + \mu_{211}(t) + \mu_{212}(t))] \notag \\
i_{13}(t) &amp;=  [(\mu_{111}(t) + \mu_{121}(t) + \mu_{212}(t) + \mu_{222}(t)) – (\mu_{112}(t) + \mu_{122}(t) + \mu_{211}(t) + \mu_{221}(t))] \notag \\
i_{23}(t) &amp;=  [(\mu_{111}(t) + \mu_{122}(t) + \mu_{211}(t) + \mu_{222}(t)) – (\mu_{112}(t) + \mu_{121}(t) + \mu_{212}(t) + \mu_{221}(t))] \notag \\
i_{123}(t) &amp;=  [(\mu_{111}(t) + \mu_{122}(t) + \mu_{212}(t) + \mu_{122}(t)) – (\mu_{112}(t) + \mu_{121}(t) + \mu_{211}(t) + \mu_{222}(t))]
\tag{3.5}
\end{align}\]</span>
<p>Each genotype can draw a growth curve using its growth parameters (a, b, r) estimated from raw data, from which we can chart the curves of pairwise and three-way epistatic effects using equation (4). Three markers AATTC_nn_np_2815, AATTC_lm_ll_3034 and AATTC_nn_np_1615 that produce a significant three-way interaction for parameter x of shoot length display pronounced differences in growth curve (<a href="highorder.html#fig:growth-compare">3.1</a>). The epistasis of low- and high-order performs differently to affect growth form, with three-way interactions playing a more remarkable role than pairwise epistasis (<a href="highorder.html#fig:epistasis-compare">3.2</a>).</p>
</div>
</div>
<div id="discussion-1" class="section level2">
<h2><span class="header-section-number">3.4</span> Discussion</h2>
<p>Genetic interactions have been thought to contribute to a significant portion of genetic variance for quantitative traits of critical importance to evolutionary biology, agriculture and medicine (Nelson et al. 2013; Mackay 2014). While pairwise interactions have been a major focus of quantitative genetic studies, there has been growing evidence that genetic interactions involving three or more loci play an important role in affecting the phenotypic differentiation of traits (Wang et al. 2010; Dowell et al. 2010; Pettersson et al. 2011; Pang et al. 2013; Weinreich et al. 2013; Taylor and Ehrenreich 2014; Taylor and Ehrenreich 2015). Because of its complexity due to a network of interactions, the detection of high-order epistasis is extremely difficult (Mackay 2014). More importantly, interpretation of high-order epistasis and its contribution to overall genetic architecture can be better made by jointly analyzing all possible low- and high-order interactions among genes. This has added an extra challenge to statistical modeling and detection of this important phenomenon. Thanks to the recent development of statistical models for high-dimensional variable selection, we have reformed a statistical modeling framework for detecting high-order epistasis by focusing on three-way interactions.</p>
<p>Our model extends Hao and Zhang’s (2014) forward selection-based algorithm iFORM that has proven to be robust and efficient for computing and detecting two-way interactions between predictors (including continuous predictors). A favorable property of iFORM is its capacity to detect interactions even if the dimension of predictors is extremely high relative to a sample size used. The fundamental assumption used by iFORM is the heredity principle, i.e., the existence of interactions between a pair of variables that each has at least weak main effects. After extending it to characterize three-way interactions, this assumption can be relaxed for the third variable; i.e., even if there is no detectable main effect for the third marker, then extended iFORM can still detect the three-way interaction. This property may explain the reason why high-order epistatic model outperforms low-order epistatic model, as demonstrated from the detection of significant genetic interactions in a real data of a woody plant, mei (Prunus mume). It was found from a recent study that loci participating in high-order genetic interactions may not individually have measurable effects (Bloom et al. 2013). As a result, our model can be used as a general tool to detect genetic interactions of various orders and, therefore, elucidate the overall picture of genetic architecture by capturing the so-called missing heritability.</p>
<p>The model was investigated by simulation studies whose result help users to determine an optimal design of mapping or association studies in terms of sample size, phenotyping precision and the number of markers. Its application to P. mume genetic mapping leads to the detection of key loci and their interactions expressed at the low- and high-order levels for the growth form of shoots. …. the R packages was created and made available through CRAN (Comprehensive R Archive Network)[<a href="https://cran.r-project.org/" class="uri">https://cran.r-project.org/</a>]. We packed iFORM/eQTL in R with the source code is available at Center for Statistical Genetics (website)[<a href="http://statgen.psu.edu/software/" class="uri">http://statgen.psu.edu/software/</a>]</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="highdeqtl.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="iform-functional-mapping-a-computational-method.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-HighOrderEpistasis.Rmd",
"text": "Edit"
},
"download": ["Defense.pdf", "Defense.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
