<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Dissertation Defense</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook.">
  <meta name="generator" content="bookdown 0.3.8 and GitBook 2.6.7">

  <meta property="og:title" content="Dissertation Defense" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Dissertation Defense" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Kirk Gosik">


<meta name="date" content="2017-02-12">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="high-order-epistatic-networks.html">
<link rel="next" href="conclusions.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>
<li><a href="...">The thesis proposal of Kirk Gosik reviewed by the following</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html"><i class="fa fa-check"></i><b>2</b> High Dimensional eQTL</a><ul>
<li class="chapter" data-level="2.1" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#motivation"><i class="fa fa-check"></i><b>2.1</b> Motivation</a></li>
<li class="chapter" data-level="2.2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#methods"><i class="fa fa-check"></i><b>2.2</b> Methods</a><ul>
<li class="chapter" data-level="2.2.1" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#experimental-design"><i class="fa fa-check"></i><b>2.2.1</b> Experimental design</a></li>
<li class="chapter" data-level="2.2.2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#adaptation-of-iform-procedure"><i class="fa fa-check"></i><b>2.2.2</b> Adaptation of iFORM procedure</a></li>
<li class="chapter" data-level="2.2.3" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#iform"><i class="fa fa-check"></i><b>2.2.3</b> iFORM</a></li>
<li class="chapter" data-level="2.2.4" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#some-considerations"><i class="fa fa-check"></i><b>2.2.4</b> Some considerations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#application"><i class="fa fa-check"></i><b>2.3</b> Application</a><ul>
<li class="chapter" data-level="2.3.1" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#simulation-results"><i class="fa fa-check"></i><b>2.3.1</b> Simulation Results</a></li>
<li class="chapter" data-level="2.3.2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#real-data-analysis"><i class="fa fa-check"></i><b>2.3.2</b> Real Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#discussion"><i class="fa fa-check"></i><b>2.4</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html"><i class="fa fa-check"></i><b>3</b> High-order Epistatic Networks</a><ul>
<li class="chapter" data-level="3.1" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#motivation-1"><i class="fa fa-check"></i><b>3.1</b> Motivation</a></li>
<li class="chapter" data-level="3.2" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#methods-1"><i class="fa fa-check"></i><b>3.2</b> Methods</a><ul>
<li class="chapter" data-level="3.2.1" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#mapping-and-association-studies"><i class="fa fa-check"></i><b>3.2.1</b> Mapping and association studies</a></li>
<li class="chapter" data-level="3.2.2" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#epistatic-model"><i class="fa fa-check"></i><b>3.2.2</b> Epistatic model</a></li>
<li class="chapter" data-level="3.2.3" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#iform-with-high-order-epistasis"><i class="fa fa-check"></i><b>3.2.3</b> iForm with high-order epistasis</a></li>
<li class="chapter" data-level="3.2.4" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#theoretical-properties"><i class="fa fa-check"></i><b>3.2.4</b> Theoretical Properties</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#application-1"><i class="fa fa-check"></i><b>3.3</b> Application</a><ul>
<li class="chapter" data-level="3.3.1" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#simulation-studies"><i class="fa fa-check"></i><b>3.3.1</b> Simulation Studies</a></li>
<li class="chapter" data-level="3.3.2" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#worked-example"><i class="fa fa-check"></i><b>3.3.2</b> Worked Example</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#discussion-1"><i class="fa fa-check"></i><b>3.4</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html"><i class="fa fa-check"></i><b>4</b> iForm Functional Mapping (A computational method)</a><ul>
<li class="chapter" data-level="4.1" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#motivation-2"><i class="fa fa-check"></i><b>4.1</b> Motivation</a></li>
<li class="chapter" data-level="4.2" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#methods-2"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#functional-mapping"><i class="fa fa-check"></i><b>4.2.1</b> Functional Mapping</a></li>
<li class="chapter" data-level="4.2.2" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#regression-by-linear-combination-of-basis-functions"><i class="fa fa-check"></i><b>4.2.2</b> Regression by linear combination of basis functions</a></li>
<li class="chapter" data-level="4.2.3" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#legendre-polynomials"><i class="fa fa-check"></i><b>4.2.3</b> Legendre Polynomials</a></li>
<li class="chapter" data-level="4.2.4" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#model"><i class="fa fa-check"></i><b>4.2.4</b> Model</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#application-2"><i class="fa fa-check"></i><b>4.3</b> Application</a><ul>
<li class="chapter" data-level="4.3.1" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#simulation-studies-1"><i class="fa fa-check"></i><b>4.3.1</b> Simulation Studies</a></li>
<li class="chapter" data-level="4.3.2" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#worked-example-1"><i class="fa fa-check"></i><b>4.3.2</b> Worked Example</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#discussion-2"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>5</b> Conclusions</a><ul>
<li class="chapter" data-level="5.1" data-path="conclusions.html"><a href="conclusions.html#summary"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="conclusions.html"><a href="conclusions.html#discussion-3"><i class="fa fa-check"></i><b>5.2</b> Discussion</a></li>
<li class="chapter" data-level="5.3" data-path="conclusions.html"><a href="conclusions.html#future-steps"><i class="fa fa-check"></i><b>5.3</b> Future Steps</a><ul>
<li class="chapter" data-level="5.3.1" data-path="conclusions.html"><a href="conclusions.html#aim-1"><i class="fa fa-check"></i><b>5.3.1</b> Aim 1</a></li>
<li class="chapter" data-level="5.3.2" data-path="conclusions.html"><a href="conclusions.html#aim-2"><i class="fa fa-check"></i><b>5.3.2</b> Aim 2</a></li>
<li class="chapter" data-level="5.3.3" data-path="conclusions.html"><a href="conclusions.html#aim-3"><i class="fa fa-check"></i><b>5.3.3</b> Aim 3</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Dissertation Defense</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="iform-functional-mapping-a-computational-method" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> iForm Functional Mapping (A computational method)</h1>
<div id="motivation-2" class="section level2">
<h2><span class="header-section-number">4.1</span> Motivation</h2>
<p><strong>Shoot-Root Paper</strong> (The genetic architecture of shoot-root covariation during seedling emergence of a dessert tree, Populus euphratica)</p>
<p>Since most traits associated with growth and development can be better described by a dynamic process (, ), it is more biologically meaningful to map these traits as growth curves (Sun and Wu, 2015). Several approaches have integrated growth equations into the likelihood of genetic mapping, leading to the birth of a so-called functional mapping model (Ma et al., 2002; Wu and Lin, 2006; Li and Sillanpaa, 2015; Muraya et al., 2016). Functional mapping allows the developmental change of genetic control to be characterized across time and space (He et al., 2010; Li and Wu, 2010). By modeling the longitudinal mean-covariance structures using a set of parsimonious parameters, functional mapping has proven of great statistical power in gene identification and the utilization of sparse phenotypic data (Hou et al., 2005, 2006). An alternative to functional mapping is to map growth QTLs by estimating growth parameters for each genotype based on growth equations and associating these parameters with markers (Wu et al., 2002).</p>
<p>From fundamental principles of biophysical and biochemical processes, logistic equations that capture different stages of organ development have been derived, which show robust biological relevance (West et al., 2001). Sun et al. (2014) dissect logistic growth curves into several key landmarks of development using the concept of heterochrony, defined as the asymptotic growth, relative growth rate, the timing of inflection and the duration of linear growth.</p>
<p><strong>from research statement</strong></p>
<p>Treating the molecular biology of an organism as a complex trait, it would be likely that the expression profiles of certain genes, proteins or other metabolites would follow a more functional or dynamic phenotype. This information could be lost or greatly limited by treating the response as a single static predictor. In an attempt to capture all relevant information available I am currently extending the procedure further to include the entire growth function as a phenotype. The mean growth curve is estimated for the sample and then orthogonal polynomials are used to assist in fitting the genetic effects for each marker or interaction between the markers. This would allow the genetic effect some flexibility over time and give a more representative fit. I anticipate submitting a third manuscript in late spring around this topic. This could also be used with other semi parametric functions to model dynamics or other types of non-linear functions that characterizes the biological systems being evaluated.</p>
<p><strong>Update</strong> (This is from Han Hao’s proposal, update for yours)</p>
<p>Changes in developmental timing and rate, named as heterochrony, have long been believed to be a major force in the evolution of phenotype (Gould, 1977; Wilson et al., 1988). It has been observed that relatively few genetic changes in heterochrony through the endocrine regulation of metamorphosis can cause profound morphological consequences (Moss, 2007). For example, although humans and chimpanzees are closely related, their skull shape and brain growth are different dramatically during early development (Rice, 2002; Mitteroecker et al., 2004; King, 2004). A recent phylogenic investigation revealed that changes in developmental timing are a crucial step for birds to evolve from dinosaurs (Bhullar, 2012). The consequence of these changes leads birds to take months to reach sexual maturity, allowing them to retain the physical characteristics of baby stages characterized by dinosaurs that take years to mature. One question that naturally arises from these evolutionary divergences is what mechanisms are implicated for heterochrony and the change of biological clock.</p>
<p>Many studies have pursued to identify the molecular control of developmental timing; mostly using Caenorhabditis elegans as an example, these studies have identified heterochronic genes that orchestrate the timing of cell divisions and fates during development regulated by microRNAs and their targets (Ambros, 2000; Rougvie, 2001; Pasquinelli and Ruvkun, 2002; Banerjee and Slack, 2002; Moss, 2007). Focusing on particular pathways causing heterochonic changes, none of these studies has provided an entire picture of the genetic control of heterochrnoy. Furthermore, the effects of heterochronic genes on the evolution of complex phenotypes have not been quantified, limiting the inference and prediction of evolutionary changes.</p>
<p>In this chapter, we develop a general framework for characterizing the genetic architecture of heterochrony based on widely used genetic mapping approaches. Genetic mapping has been proved to be powerful for mapping and studying quantitative trait loci (QTLs) involved in complex traits (Lander and Botstein, 1986), and has been increasingly integrated with network biology to better elucidate the mechanisms of the way QTL acts and interacts with other factors (Wang et al., 2012b). In the decade, genetic mapping has developed to a point at which this approach can characterize QTLs that control the process of development, leading to the birth of functional mapping (Ma et al., 2002; Wu and Lin, 2006; Li and Wu, 2010). Functional mapping implements mathematical aspects of developmental principles into a mapping framework, equipped with a capacity to study the interplay between QTLs and development. Here, we extend functional mapping to characterize QTLs controlling heterochrony (named hQTLs), specified by three parameters (i) the onset of a particular process, (ii) its offset, and (iii) the rate at which the process proceeds. By using growth equation as an example, we exemplify the procedure of model derivations as well as the practical use of the model in hQTL detection. In the end, we discuss an issue of how hQTLs can be integrated with developmental evolutionary biology (evo-devo), a fast-growing discipline of biology in the recent years.</p>
<p><strong>Update</strong></p>
<p>GENERALIZED ADDITIVE MODEL SELECTION </p>
<p>In many applications it may be too restrictive to suppose that the effect of all of the predictors is captured by a simple linear fit of the form, <span class="math inline">\(η(x_i) = \sum_{j=1}^p \beta_jx_{ij}\)</span>. Generalized additive models, introduced in , allow for greater flexibility by modeling the linear predictor of a generalized linear model as a sum of more general functions of each variable: <span class="math display">\[η(x_i) = \sum_{j=1}^p f_j(x_{ij}),\]</span> where the <span class="math inline">\(f_j\)</span> are unknown functions, assumed to be smooth or otherwise low-complexity.</p>
<p>Add a third paragraph to motivation section Continue on about the rest of the chapter as an overview</p>
</div>
<div id="methods-2" class="section level2">
<h2><span class="header-section-number">4.2</span> Methods</h2>
<div id="functional-mapping" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Functional Mapping</h3>
<p><strong>Update</strong> (This is from Han Hao’s proposal, update for yours)</p>
<p>Functional mapping is a group of methods used for mapping QTLs related to functional valued traits that are measured over a certain time period. These function-valued traits are widely seen in growth analysis, shape analysis, network analysis, and clinical trials. By integrating functional features with genetic analysis, functional mapping methods often help to increase the statistical power and the biological relevance between the detected QTLs and biological traits. Functional mapping methods were first developed using parametric curves to describe the functional valued traits, such as growth curves or pharmacodynamic models (Ma et al., 2002; Lin et al., 2005). Later, semi-parametric and non-parametric models were introduced for complex traits that do not have specific mathematical forms (Das et al., 2011, 2013).</p>
<p>The key of functional mapping methods is the modelling of both functional means and the covariance structure across measurements from the same individual. When modelling the functional means, biological background is first examined to select the best function, either parametric or non-parametric. The covariance structure is usually modeled by parsimonious and flexible approaches such as autoregressive, antedependence, or nonparametric structures. The functional means and covariance structures can be integrated into a likelihood function, then hypothesis testing can be performed with a likelihood ratio testing approach.</p>
<p><strong>Update: Shoot-Root Paper</strong> (pages 13 - 14) fitted by a three-parameter growth equation, through a nonlinear least-square approach, which is expressed as (1) where g(t) is the trait value at time t, and three parameters a, b and r have different biological meanings: a is the limit value of g when t→∞, r is the relative growth rate, and a/ (1 + b) denotes the initial value of g when t = 0. After the three growth parameters were estimated for each progeny, we further determined heterochronic parameters, i.e., the timing of inflection point (tI), the timing of maximum acceleration (ta), the timing of maximum deceleration (td), and the duration of linear growth (L) (Sun et al., 2014).</p>
<p>QTL mapping: For each progeny, we estimated a series of heterochronic parameters using equation (1) and then treated these estimates as phenotypic values to perform QTL mapping. There are two statistical approaches for QTL mapping, mixture model for sparse molecular markers and multiplicative model for dense markers. Because the linkage map constructed is quite dense, we employed the multiplicative model that assumes QTLs are located at the positions of markers. For the same heterochronic parameter expressed in shoot length and taproot length, the multiplicative likelihood model is expressed as (2) where  is the unknown parameters;$ yi = (y1i, y2i)$ is the growth parameter vector of progeny i for shoot length (coded by 1) and taproot length (coded by 2); nj is the number of progeny with SNP genotype j; and <span class="math inline">\(fj(yi)\)</span> is a bivariate normal distribution for progeny i with the expected mean vector for genotype j (1j, 2j) and the variance-covariance matrix  containing the variances (12, 22) and correlation between the two traits (). Statistical methods based on the likelihood (2) have been established to estimate the model parameters  = (1j, 2j, 12, 22, ).</p>
</div>
<div id="regression-by-linear-combination-of-basis-functions" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Regression by linear combination of basis functions</h3>
<p>** GAM Selection paper**</p>
<p>Our proposed estimator selects between fitting each fj as zero, linear, or nonlinear, as determined by the data. In so doing it retains the interpretability advantages of linear fits when appropriate, while capturing strong non-linear relationships when they are present. Our method, which we call GAMSEL (Generalized Additive Model Selection), is based on optimizing a penalized (negative) log-likelihood criterion of the form</p>
<p><strong>Regression by linear combination of basis functions. by Risi Kondor</strong></p>
<p>Solving for the θ’s To find the optimal value for <span class="math inline">\(θ_0, θ_1, . . . , θ_P\)</span> we 1. define a loss function L; 2. using the loss function define the empirical risk <span class="math inline">\(Remp(θ)\)</span> quantifying the loss over all the training data for particular values of <span class="math inline">\(θ0, θ1, . . . , θP\)</span> ; 3. solve for the particular setting of the parameters (denoted <span class="math inline">\(θ_0, θ_1, . . . , θ_P\)</span>) that minimizes the empirical risk.</p>
<p>We shall use the squared error loss function <span class="math display">\[L(y, f(x)) =\frac{1}{2}(y − f(x))^2\]</span></p>
</div>
<div id="legendre-polynomials" class="section level3">
<h3><span class="header-section-number">4.2.3</span> Legendre Polynomials</h3>
<div class="figure" style="text-align: center"><span id="fig:Legendre-fig"></span>
<img src="Defense_files/figure-html/Legendre-fig-1.png" alt="First 10 Legendre Polynomials" width="80%" />
<p class="caption">
FIGURE 4.1: First 10 Legendre Polynomials
</p>
</div>
<p><strong>From Wikipedia</strong> An important property of the Legendre polynomials is that they are orthogonal with respect to the L2-norm on the interval −1 ≤ x ≤ 1:</p>
<span class="math display" id="eq:var-beta">\[\begin{equation} 
\begin{split}
\mathrm{Var}(\hat{\beta}) &amp; =\mathrm{Var}((X&#39;X)^{-1}X&#39;y)\\
 &amp; =(X&#39;X)^{-1}X&#39;\mathrm{Var}(y)((X&#39;X)^{-1}X&#39;)&#39;\\
 &amp; =(X&#39;X)^{-1}X&#39;\mathrm{Var}(y)X(X&#39;X)^{-1}\\
 &amp; =(X&#39;X)^{-1}X&#39;\sigma^{2}IX(X&#39;X)^{-1}\\
 &amp; =(X&#39;X)^{-1}\sigma^{2}
\end{split}
\tag{4.1}
\end{equation}\]</span>
<p><span class="math display">\[P_n(x) = \frac{1}{2^n}\sum_{k=0}^{n}{{n}\choose{k}}^2(x-1)^{n-k}(x+1)^k\]</span></p>
<span class="math display">\[\sum_{k=0}^{n}{{n}\choose{k}}{{-n-1}\choose{k}}{\left(\frac{1-x}{2}\right)}^k\]</span>
<span class="math display" id="eq:leg-eq">\[\begin{equation}
\begin{split}
P_n(x) &amp; = \frac{1}{2^n}\sum_{k=0}^{n}{{n}\choose{k}}^2(x-1)^{n-k}(x+1)^k \\
&amp; = \sum_{k=0}^{n}{{n}\choose{k}}{{-n-1}\choose{k}}{\left(\frac{1-x}{2}\right)}^k \\
&amp; = 2^{-n}\sum_{k=0}^{n} x^k {{n}\choose{k}}{{\frac{n+k+1}{2}}\choose{k}} \\
\end{split}
\tag{4.2}
\end{equation}\]</span>
<p><strong>update</strong> 2HIGWAS </p>
<p>We consider the complex trait as a growth trait. Thus, it is biologically meaningful to implement a growth equation, like a logistic curve, to describe growth trajectory . We describe the population mean growth curve by the growth equation <span class="math display">\[ \mu(t) = a/(1 + b * exp(-r*t)) \]</span></p>
<p>where a, b and r are growth parameters each of biological interpretation, with a being the asymptotic growth, b being the initial amount of growth and r being the relative growth rate. Timevarying additive and dominant effects of significant SNPs are modeled by a nonparametric approach, such as Legendre orthogonal polynomial used in quantitative genetic studies (,], expressed as</p>
<p><span class="math display">\[ \alpha_j(t) = (L_0(t), L_1(t), ... , L_s(t))*(u_{j0}, u_{j1},...,u_{js})^T \]</span> <span class="math display">\[ \beta_j(t) = (L_0(t), L_1(t), ... , L_{s&#39;}(t))*(v_{j0}, v_{j1},...,v_{js&#39;})^T \]</span></p>
<p>where <span class="math inline">\(L_0(t), L_1(t), ... , L_s(t)\)</span> and <span class="math inline">\(L_0(t), L_1(t), ... , L_{s&#39;}(t)\)</span> are the LOP of orders <span class="math inline">\(s\)</span> and <span class="math inline">\(s&#39;\)</span>, respectively; and <span class="math inline">\(u_{j0}, u_{j1},...,u_{js}\)</span> and <span class="math inline">\(v_{j0}, v_{j1},...,v_{js&#39;}\)</span> are the vectors of time-invariant additive and dominant effects, respectively. Orders <span class="math inline">\(s\)</span> and <span class="math inline">\(s&#39;\)</span>, selected from information criteria, such as Akaike information criterion (AIC) and Bayesian information criterion (BIC), are usually much smaller than M so that the dimension of response phenotypic data is reduced through LOP modeling. The variance maxtrix of residual errors is assumed to follow the first-order structured antedependence [SAD(1)] process (). The SAD(1) model has been used in previous growth modeling studies (, , ).</p>
<p><strong>free write</strong></p>
<p>Because the non-parametic nature of the legendre orthogonal polynomials, it was advantages for both dimension reduction and also handling unevenly spaced, missing or non-uniform time measurements from different subjects in the dataset.</p>
<p>The search performed by the procedure checks and uses the legendre polynomials of different orders to find the best fitting form of the genetic variation from the mean curve for each of the genotypes or epistasis between genotypes considered in the model.</p>
</div>
<div id="model" class="section level3">
<h3><span class="header-section-number">4.2.4</span> Model</h3>
<p><strong>2HIGWAS</strong> Because interaction effects are considered, the coverage probability P is redefined as three quantities: PA, the coverage rate of all true predictors; PM, the coverage rate of all main effects; and PI, the coverage rate of all interaction effects.</p>
<p><strong>BIC</strong></p>
<p>Similarly, following the classical BIC rule (Swartz 1978), we can define the BIC rule for the cubic MESS model as</p>
<p><span class="math display">\[BIC(\lambda, \lambda_v ) = -2*LogLik + log(n)*(df + df_v)\]</span> n is the number of subjects  <span class="math display">\[\lambda = smoothing paramter\]</span> <span class="math display">\[\lambda_v = smoothing paramter, number of knots in the cubic spline\]</span></p>
<p>Considering a comprehensive gentotyping for n progeny each measure across different time points.</p>
</div>
</div>
<div id="application-2" class="section level2">
<h2><span class="header-section-number">4.3</span> Application</h2>
<div id="simulation-studies-1" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Simulation Studies</h3>
<p>extensive simulation studies were performed to ascertain the validity of the model. Computational verified by cross validation, bootstrapping. Having a training and a testing set to help provide some insight on false positive rates, selecting the correct fit of the polynomial for the genetic and epsitatic effects.</p>
<ul>
<li>rates at which correct markers/epistasis was selected?</li>
<li>rates at which the correct order polynomial was fit for the correct marker/epistasis</li>
</ul>
</div>
<div id="worked-example-1" class="section level3">
<h3><span class="header-section-number">4.3.2</span> Worked Example</h3>
<p>Same mei tree dataset used in previous chapter to see if any new discoveries can be made by incorporating the time component and fitting the growth parameters simulatneously throughout the procedure.</p>
<p>if it doesn’t take too much time (maybe by the time defense comes around) cross validate the selection proceudre by leaving out samples and running model again. How many times do you get the same results? LOOCV tables</p>
<p>Also consider with holding predictors from the dataset to see if the selection is changed at all. Start with random ones and then maybe strategically select important predictors to remove to see how it affects the fit</p>
</div>
</div>
<div id="discussion-2" class="section level2">
<h2><span class="header-section-number">4.4</span> Discussion</h2>
<ul>
<li>Downfalls or areas of concern for fitting the genetic effects to the polynomials?</li>
<li>How do we know there is epistasis?</li>
<li>how do you keep it from over selecting</li>
<li>what if it doesn’t follow the heredity or hierarchy principles?</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="high-order-epistatic-networks.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="conclusions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-iFormFunctionalMapping.Rmd",
"text": "Edit"
},
"download": ["Defense.pdf", "Defense.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
