<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta name="description" content="My Dissertation Written in Bookdown.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My Dissertation Written in Bookdown." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  <meta name="twitter:description" content="My Dissertation Written in Bookdown." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  

<link rel="next" href="highdeqtl.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css, toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

true

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:title:end-->
<!--bookdown:title:start-->
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p> </p>
<div id="background" class="section level2">
<h2><span class="header-section-number">1.1</span> Background</h2>
<p>
There are several techniques used for studying genetics and mapping the results. Some of the more popular techniques include cross-breeding experiments or, in the case of humans, the examination of family histories, known as pedigrees. More recently, CRISPR/Cas9 can be used to mimic mitotic recombination to help map out genes as well. ()
</p>
<p>
Construction of genetic maps are a variety of techniques used to show relative positions between genes or other sequence features of the genome and the phenotype that is controlled by such sequences. Genes are very useful markers but they are by no means ideal. One problem, especially with larger genomes such as those of vertebrates and flowering plants, is that a map based entirely on genes is not very detailed.() Genes have long areas of non-coding regions between them and therefore result in large gaps from gene to gene. This is further complicated because not every gene has allelic forms that can be easily or conveniently distinguished. With these considerations in mind gene maps may not be comprehensive enough and other markers may be needed.
</p>
<p>
According to brown, mapped features that are not genes are called DNA markers. As with gene markers, a DNA marker must have at least two alleles to be useful. There are three types of DNA sequence feature that satisfy this requirement: restriction fragment length polymorphisms (RFLPs), simple sequence length polymorphisms (SSLPs), and single nucleotide polymorphisms (SNPs). () The genetic markers that have been emphasized in this work are single nucleotide polymorphisms. Attempting to be at the highest levels of resolution for identifying quantitative traits, using SNPs are the most specific case. This will give exact location of the nucleotide that may be impacting the genetic control over the phenotype.
</p>
<div class="figure" style="text-align: center"><span id="fig:Snp-Pic"></span>
<img src="images/SNP_Picture.png" alt="SNP Picture" width="80%" />
<p class="caption">
FIGURE 1.1: SNP Picture
</p>
</div>
<p>
There are several goals to genetic mapping and association studies that identify certain regions of the genome that contain genes involved in specifying a quantitative trait, referred to as quantitative trait loci (QTLs). One main goal is to estimate the genetic effects of these loci. The relationship between the genetic effects of QTLs and the phenotypic value of quantitative traits can be described by a linear model (, ). Typically, because of the high throughput nature of the data there are a large number of markers across the whole genome, and most of the markers may have very little or next no effect on the phenotype under study. The models can be very sparse, with most cases, the number of genetic markers or variables is bigger than the sample size, especially when interactions among markers are considered. This makes a model is over saturated and further model selection techniques may be required to capture the necessary information. 
</p>
<div class="figure" style="text-align: center"><span id="fig:system-map"></span>
<img src="images/SystemsMapping.png" alt="Systems Map" width="80%" />
<p class="caption">
FIGURE 1.2: Systems Map
</p>
</div>
</div>
<div id="some-exisiting-methods" class="section level2">
<h2><span class="header-section-number">1.2</span> Some Exisiting Methods</h2>
<p>
Numerous methods exist and are being developed to measure and find quantitative trait loci (QTL) effects. These methods can broadly fall into three main categories. These categories are Least-Square methods, maximum likelihood and Bayesian approaches. () Each method has advantages and considerations that you would need to be aware before conducting analyses to find QTL effects from the given markers. A brief discussions on a few of the methods are given to highlight some areas of consideration and how the methods proposed can handle such considerations.
</p>
<p>
Marker Regression would fall in the category of Least Squares approaches. If looking at one marker analysis general t-test and ANOVA procedures can be used to analyze the relationship. It is not recommended however for use in general practice because you do not know how dense the markers are measured. QTL interval mapping would be preferred in such an analysis because the methods take account for missing genotype data that may not have been measured. When estimating a QTL position through maximum likelihood methods, like interval mapping, positions of other possible QTLs could affect the detection of the true position. Neighboring QTLs could possibly flatten the likelihood in instances where there are multiple QTLs on the same chromosome. This would make an effect look less significant at a given location than it actually is. Another possibility is that in the search over the interval you may find an area where the likelihood could reach a peak but could be a “ghost” QTL. This is where an effect is observed because a neighboring QTL is skewing the results at the particular position you are looking in and the result is a false discovery of the position. Marker Regression has been shown to improve interval mapping, which is call Composite Interval Mapping. This is where the QTL position found is also combined in a linear regression where the covariates are the other markers in the dataset. By including the markers as covariates the other position in the chromosome are accounted for in the analysis and false discovery is reduced.
</p>
<p>
The analysis of interval mapping and single marker analyses has shown to be effective but it limits our inference to one marker at a time as a possible loci that controls a trait. Using Marker Regression however you can incorporate multiple markers in a single analysis to test for possible QTL for a given trait. It is cautioned that running such an analysis is only an approximate test because the null hypothesis is there is no difference between the marker levels and therefore a non-mixture distribution but the alternative is a mixture of distributions. The assumptions regression would make of the errors within the marker type to be normally distributed may not be entirely met if the QTL’s fall between the marker regions. However  have shown that a direct regression of phenotypes on marker types, provides the same information about location of QTL-effects without having to step to all positions on the interval. With this information using the entire marker set in a regression analysis would provide a nice, computationally efficient way to map out the genetic architecture of a trait.
</p>
</div>
<div id="chapter-overview" class="section level2">
<h2><span class="header-section-number">1.3</span> Chapter Overview</h2>
<p>
The main theme of this paper is to propose an improvements on selection procedures which use regression techniques to approach high dimensional variable selection such as the ones arising in epistatic analysis. The variable selection procedure for QTL mapping can be seen as one of deciding which subset of variables have effects on the phenotypes of interest, and identifying those specific traits out of all possible effects between the markers. Each procedure proposed is a forward selecting method that starts with the empty set. After starting with the empty set each procedure continues to add markers to the model set as possible QTLs. Once a designated stopping criteria is met the final model is fit and the effects are estimated. Each of the next three chapters focuses on a new selection procedure and the properties of them.
</p>
<div id="highdeqtl" class="section level3">
<h3><span class="header-section-number">1.3.1</span> HighDeQTL</h3>
<p>
In this chapter we introduce the iform procedure, originally proposed by (). Included in the chapter is the algorithm and how it compares to forward selection. From there it is adapted to use for a genetic mapping studies. The properties of this will be explored in detail. Simulation studies were conducted to assess how well the properties are met and under what conditions. Comparisons to other models was also explored to get a sense of the utility and advantages that come with the new selection procedure. After the comparisons and simulations a real world application is performed. The application was a reanalysis of a data set using C Elegans first proposed by .
</p>
</div>
<div id="higher-order-epistasis" class="section level3">
<h3><span class="header-section-number">1.3.2</span> Higher Order Epistasis</h3>
<p>
The next chapter considers higher order epistasis and its importance. Currently it is under studied because of practical limitations not because of biological limitations or relevance. The iFORM procedure proposed in Chapter 2 is then extended to incorporate higher orders of epistatic effects. Even though similar methods are being used the properties are still studied and assessed. Simulation studies were performed to assess practical applications. Several scenarios and comparison models were considered to extensively look at what properties were being met and which were not. Then an application to Mei tree growth was conducted in order to see the real world application of such a selection procedure. Different growth parameters were previously fit and these were used as the phenotype. Interesting and more predictive implications came out of the model when considering higher order epistasis throughout the selection procedure.
</p>
</div>
<div id="iform-funcional-mapping" class="section level3">
<h3><span class="header-section-number">1.3.3</span> iForm Funcional Mapping</h3>
<p>
The application of the Mei tree growth led into exploring to use a more functional phenotype as a response throughout the selection procedure. In order to use all relevant information of the repeated measure data, it would take an additional computation burden to the selection and the modeling but it would also give more power and flexibility to the modeling that would not be present otherwise. It is important to use all relevant information in order to make the most accurate prediction about the data. Using a growth curve model to assist in fitting the data would help ease some of the computational burden. The selection of genetic effects however would be very simplistic as some additive shift to the curve. This view may not be the most accurate and therefore more complicated structures will be implemented to produce better results. Legendre orthogonal polynomials were considered to model the genetic effects. These polynomials have various forms and would allow for the genetic effect to follow different patterns but also not induce unnecessary correlation between predictors when included in the model. Again, simulation studies were conducted and a reanalysis of the full Mei tree dataset was considered.
</p>
</div>
<div id="conclusions" class="section level3">
<h3><span class="header-section-number">1.3.4</span> Conclusions</h3>
<p>
Finally in the last chapter it will focus on comparisons and discussion around the models proposed. Future aims of the research goals will be explored also. Seeing how my current research will lead into the aims and what possible directions can be explored with the given statistical frameworks.
</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="highdeqtl.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-Introduction.Rmd",
"text": "Edit"
},
"download": ["Defense.pdf", "Defense.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
