<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Dissertation Defense</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="Dissertation Defense" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Dissertation Defense" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Kirk Gosik">


<meta name="date" content="2017-02-13">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="high-dimensional-eqtl.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>
<li><a href="...">The thesis proposal of Kirk Gosik reviewed by the following</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html"><i class="fa fa-check"></i><b>2</b> High Dimensional eQTL</a><ul>
<li class="chapter" data-level="2.1" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#motivation"><i class="fa fa-check"></i><b>2.1</b> Motivation</a></li>
<li class="chapter" data-level="2.2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#methods"><i class="fa fa-check"></i><b>2.2</b> Methods</a><ul>
<li class="chapter" data-level="2.2.1" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#experimental-design"><i class="fa fa-check"></i><b>2.2.1</b> Experimental design</a></li>
<li class="chapter" data-level="2.2.2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#adaptation-of-iform-procedure"><i class="fa fa-check"></i><b>2.2.2</b> Adaptation of iFORM procedure</a></li>
<li class="chapter" data-level="2.2.3" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#iform"><i class="fa fa-check"></i><b>2.2.3</b> iFORM</a></li>
<li class="chapter" data-level="2.2.4" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#some-considerations"><i class="fa fa-check"></i><b>2.2.4</b> Some considerations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#application"><i class="fa fa-check"></i><b>2.3</b> Application</a><ul>
<li class="chapter" data-level="2.3.1" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#simulation-results"><i class="fa fa-check"></i><b>2.3.1</b> Simulation Results</a></li>
<li class="chapter" data-level="2.3.2" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#real-data-analysis"><i class="fa fa-check"></i><b>2.3.2</b> Real Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="high-dimensional-eqtl.html"><a href="high-dimensional-eqtl.html#discussion"><i class="fa fa-check"></i><b>2.4</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html"><i class="fa fa-check"></i><b>3</b> High-order Epistatic Networks</a><ul>
<li class="chapter" data-level="3.1" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#motivation-1"><i class="fa fa-check"></i><b>3.1</b> Motivation</a></li>
<li class="chapter" data-level="3.2" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#methods-1"><i class="fa fa-check"></i><b>3.2</b> Methods</a><ul>
<li class="chapter" data-level="3.2.1" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#mapping-and-association-studies"><i class="fa fa-check"></i><b>3.2.1</b> Mapping and association studies</a></li>
<li class="chapter" data-level="3.2.2" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#epistatic-model"><i class="fa fa-check"></i><b>3.2.2</b> Epistatic model</a></li>
<li class="chapter" data-level="3.2.3" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#iform-with-high-order-epistasis"><i class="fa fa-check"></i><b>3.2.3</b> iForm with high-order epistasis</a></li>
<li class="chapter" data-level="3.2.4" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#theoretical-properties"><i class="fa fa-check"></i><b>3.2.4</b> Theoretical Properties</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#application-1"><i class="fa fa-check"></i><b>3.3</b> Application</a><ul>
<li class="chapter" data-level="3.3.1" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#simulation-studies"><i class="fa fa-check"></i><b>3.3.1</b> Simulation Studies</a></li>
<li class="chapter" data-level="3.3.2" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#worked-example"><i class="fa fa-check"></i><b>3.3.2</b> Worked Example</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="high-order-epistatic-networks.html"><a href="high-order-epistatic-networks.html#discussion-1"><i class="fa fa-check"></i><b>3.4</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html"><i class="fa fa-check"></i><b>4</b> iForm Functional Mapping (A computational method)</a><ul>
<li class="chapter" data-level="4.1" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#motivation-2"><i class="fa fa-check"></i><b>4.1</b> Motivation</a></li>
<li class="chapter" data-level="4.2" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#methods-2"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#functional-mapping"><i class="fa fa-check"></i><b>4.2.1</b> Functional Mapping</a></li>
<li class="chapter" data-level="4.2.2" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#regression-by-linear-combination-of-basis-functions"><i class="fa fa-check"></i><b>4.2.2</b> Regression by linear combination of basis functions</a></li>
<li class="chapter" data-level="4.2.3" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#legendre-polynomials"><i class="fa fa-check"></i><b>4.2.3</b> Legendre Polynomials</a></li>
<li class="chapter" data-level="4.2.4" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#model"><i class="fa fa-check"></i><b>4.2.4</b> Model</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#application-2"><i class="fa fa-check"></i><b>4.3</b> Application</a><ul>
<li class="chapter" data-level="4.3.1" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#simulation-studies-1"><i class="fa fa-check"></i><b>4.3.1</b> Simulation Studies</a></li>
<li class="chapter" data-level="4.3.2" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#worked-example-1"><i class="fa fa-check"></i><b>4.3.2</b> Worked Example</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="iform-functional-mapping-a-computational-method.html"><a href="iform-functional-mapping-a-computational-method.html#discussion-2"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>5</b> Conclusions</a><ul>
<li class="chapter" data-level="5.1" data-path="conclusions.html"><a href="conclusions.html#summary"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="conclusions.html"><a href="conclusions.html#discussion-3"><i class="fa fa-check"></i><b>5.2</b> Discussion</a></li>
<li class="chapter" data-level="5.3" data-path="conclusions.html"><a href="conclusions.html#future-steps"><i class="fa fa-check"></i><b>5.3</b> Future Steps</a><ul>
<li class="chapter" data-level="5.3.1" data-path="conclusions.html"><a href="conclusions.html#aim-1"><i class="fa fa-check"></i><b>5.3.1</b> Aim 1</a></li>
<li class="chapter" data-level="5.3.2" data-path="conclusions.html"><a href="conclusions.html#aim-2"><i class="fa fa-check"></i><b>5.3.2</b> Aim 2</a></li>
<li class="chapter" data-level="5.3.3" data-path="conclusions.html"><a href="conclusions.html#aim-3"><i class="fa fa-check"></i><b>5.3.3</b> Aim 3</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Dissertation Defense</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>
There are several techniques used for studying genetics and mapping the results. Some of the more populor techniques include cross-breeding experiments or, in the case of humans, the examination of family histories, known as pedigrees. More recently, CRISPR/Cas9 can be used to mimic mitotic recombination to help map out genes as well. ()
</p>
<p>
Construction of genetic maps are a variety of techniques used to show relative positions between genes or other sequence features of the genome and the phenotype that is controlled by such sequences. Genes are very useful markers but they are by no means ideal. One problem, especially with larger genomes such as those of vertebrates and flowering plants, is that a map based entirely on genes is not very detailed.() Genes have long areas of non-coding regions between them and therefore result in large gaps from gene to gene. This is further complicated because not every gene has allelic forms that can be easily or conventiently distinguished. With these considerations in mind gene maps may not be comprehensive enough and other markers may be needed.
</p>
<p>Mapped features that are not genes are called DNA markers. As with gene markers, a DNA marker must have at least two alleles to be useful. There are three types of DNA sequence feature that satisfy this requirement: restriction fragment length polymorphisms (RFLPs), simple sequence length polymorphisms (SSLPs), and single nucleotide polymorphisms (SNPs).</p>
<p>** Varational Linear Model Paper ** </p>
<p>The goal of quantitative trait loci (QTLs) mapping and association studies is to identify certain regions of the genome that contain genes involved in specifying a quantitative trait, and to estimate the genetic effects of these loci. The relationship between the genetic effects of QTLs and the phenotypic value of quantitative traits can be described by a linear model [1, 2]. There are usually a large number of markers across the whole genome, and most of the markers have very little or even no effect on the phenotypes. So the model is sparse, and in most cases, the number of markers or variables is bigger than the sample size, especially when interactions among markers are considered. Therefore, the model is oversaturated and usually solved by shrinkage or variable selection methods. The variable selection procedure for QTLs mapping can be seen as one of deciding which subset of variables have effects on phenotypes, and identifying out all possible effects of those markers. There are many ways of variable selection. For instance, the indicator selection uses a spike (probability distribution of variables not included in model) and slab (probability distribution of variables included in model) prior to indicate whether each variable is included in one model. Such methods usually adopt Gibbs sampling or importance sampling to calculate the weighted average as the inferred results [3-5]. Shrinkage methods do not use indicators to induce the sparseness of variables, but instead specify a priori directly on variables to approximate the “slab” and “spike” shapes and use a penalty function to shrink most variables toward zeros. Early Bayesian shrinkage methods employed Markov chain Monte Carlo (MCMC) techniques to infer the models [6], which were computationally intensive. Xu et al. proposed an empirical Bayes method that used a carefully chosen prior distribution for the variables, which required much less</p>
<ol start="4" style="list-style-type: decimal">
<li>Conclusions and discussion The main goal of this paper is to propose an improved variational linear regression approach for high scale variable selection problems such as the ones arising in epistatic analysis. Variational approximation is a classic method to solve linear model, but its accuracy is limited in the case of there are many more variables than samples. Several improved methods were proposed for the variational approximation, such as hierarchical shrinkage, importance sampling and so on. But these methods still lack of efficiency or accuracy. We proposed a simple but effective method to improve the performance of VLR by dynamically deleting some specific variables, which can greatly improve the accuracy of VLR. The reason that RCVLR is more accurate than VB-EBL, VB-BAS, VB-BL may lie in that it can derive the approximate marginal distributions of all variables in a single process, while hierarchical shrinkage infers each variable separately, therefore some correlation information among variables is lost. The VI-IMP considers the correlation among variables, but it uses important sampling and a weighted average method to improve the accuracy; therefore the computational cost is higher than other variational approaches. RCVLR can obtain good performance in general, and is not sensitive to the choice of hyper parameters, while some other variational approaches, such as VB-EBL, rely heavily on selecting appropriate parameters to get good performance [18]. VI-IMP averages the inferred results under different prior combinations. But how to determine the prior ranges is a problem unsolved [4]. If an inappropriate range is inputted, the performance may be poor. But if a very wide range is set to cover the exact parameters, the algorithm will need many more samples and thus need much more computational time.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:system-map"></span>
<img src="images/SystemsMapping.png" alt="Systems Map" width="80%" />
<p class="caption">
FIGURE 1.1: Systems Map
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:Snp-Pic"></span>
<img src="images/SNP_Picture.png" alt="SNP Picture" width="80%" />
<p class="caption">
FIGURE 1.2: SNP Picture
</p>
</div>
<p>You can label chapter and section titles using <code>{#label}</code> after them, e.g., we can reference Chapter <a href="intro.html#intro">1</a>. If you do not manually label them, there will be automatic labels anyway, e.g., Chapter <a href="high-dimensional-eqtl.html#methods">2.2</a>.</p>
<p>Figures and tables with captions will be placed in <code>figure</code> and <code>table</code> environments, respectively.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, .<span class="dv">1</span>, .<span class="dv">1</span>))
<span class="kw">plot</span>(pressure, <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>, <span class="dt">pch =</span> <span class="dv">19</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:nice-fig"></span>
<img src="Defense_files/figure-html/nice-fig-1.png" alt="Here is a nice figure!" width="80%" />
<p class="caption">
FIGURE 1.3: Here is a nice figure!
</p>
</div>
<p>Reference a figure by its code chunk label with the <code>fig:</code> prefix, e.g., see Figure <a href="intro.html#fig:nice-fig">1.3</a>. Similarly, you can reference tables generated from <code>knitr::kable()</code>, e.g., see Table <a href="intro.html#tab:nice-tab">1.1</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
  <span class="kw">head</span>(iris, <span class="dv">20</span>), <span class="dt">caption =</span> <span class="st">&#39;Here is a nice table!&#39;</span>,
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>
)</code></pre></div>
<table>
<caption><span id="tab:nice-tab">TABLE 1.1: </span>Here is a nice table!</caption>
<thead>
<tr class="header">
<th align="right">Sepal.Length</th>
<th align="right">Sepal.Width</th>
<th align="right">Petal.Length</th>
<th align="right">Petal.Width</th>
<th align="left">Species</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">5.1</td>
<td align="right">3.5</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.9</td>
<td align="right">3.0</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.7</td>
<td align="right">3.2</td>
<td align="right">1.3</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.6</td>
<td align="right">3.1</td>
<td align="right">1.5</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.0</td>
<td align="right">3.6</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.4</td>
<td align="right">3.9</td>
<td align="right">1.7</td>
<td align="right">0.4</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.6</td>
<td align="right">3.4</td>
<td align="right">1.4</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.0</td>
<td align="right">3.4</td>
<td align="right">1.5</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.4</td>
<td align="right">2.9</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.9</td>
<td align="right">3.1</td>
<td align="right">1.5</td>
<td align="right">0.1</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.4</td>
<td align="right">3.7</td>
<td align="right">1.5</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.8</td>
<td align="right">3.4</td>
<td align="right">1.6</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.8</td>
<td align="right">3.0</td>
<td align="right">1.4</td>
<td align="right">0.1</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.3</td>
<td align="right">3.0</td>
<td align="right">1.1</td>
<td align="right">0.1</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.8</td>
<td align="right">4.0</td>
<td align="right">1.2</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.7</td>
<td align="right">4.4</td>
<td align="right">1.5</td>
<td align="right">0.4</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.4</td>
<td align="right">3.9</td>
<td align="right">1.3</td>
<td align="right">0.4</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.1</td>
<td align="right">3.5</td>
<td align="right">1.4</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.7</td>
<td align="right">3.8</td>
<td align="right">1.7</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.1</td>
<td align="right">3.8</td>
<td align="right">1.5</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
</tbody>
</table>
<p>You can write citations, too. For example, we are using the <strong>bookdown</strong> package <span class="citation">(Xie <a href="#ref-R-bookdown">2016</a>)</span> in this sample book, which was built on top of R Markdown and <strong>knitr</strong> <span class="citation">(Xie <a href="#ref-xie2015">2015</a>)</span>.</p>

</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-R-bookdown">
<p>Xie, Yihui. 2016. <em>Bookdown: Authoring Books and Technical Documents with R Markdown</em>. <a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>.</p>
</div>
<div id="ref-xie2015">
<p>Xie, Yihui. 2015. <em>Dynamic Documents with R and Knitr</em>. 2nd ed. Boca Raton, Florida: Chapman; Hall/CRC. <a href="http://yihui.name/knitr/" class="uri">http://yihui.name/knitr/</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="high-dimensional-eqtl.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-Introduction.Rmd",
"text": "Edit"
},
"download": ["Defense.pdf", "Defense.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
